

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Class 14 : ChIP-seq &mdash; Genome Informatics Workshop 0.1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  

    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:'../',
        VERSION:'0.1',
        COLLAPSE_INDEX:false,
        FILE_SUFFIX:'.html',
        HAS_SOURCE:  true
      };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
    <link rel="top" title="Genome Informatics Workshop 0.1 documentation" href="../index.html"/>
        <link rel="up" title="Class list" href="../Classes.html"/>
        <link rel="next" title="Class 15 : ChIP-seq / DNase I" href="class-15.html"/>
        <link rel="prev" title="Class 13 : BEDtools" href="class-13.html"/> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>



  <script src="//cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="../index.html" class="icon icon-home"> Genome Informatics Workshop</a>
        <form class="wy-form" action="../search.html" method="get">
  <input type="text" name="q" placeholder="Search docs" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../Syllabus.html">Syllabus (MOLB FIXME)</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal" href="../Classes.html">Class list</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="class-1.html">Class 1 : Virtual Machine (VM) and Linux Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-2.html">Class 2 : The command-line</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-3.html">Class 3 : The command-line (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-4.html">Class 4 : awk</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-5.html">Class 5 : Working in a cluster environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-6.html">Class 6 : Working in a cluster environment (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-7.html">Class 7 : Python : Basics</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-8.html">Class 8 : Python : Basics (2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-9.html">Class 9 : Python Concepts</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-10.html">Class 10 : Applied Python</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-11.html">Class 11 : Python Idioms</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-12.html">Class 12 : The ENCODE Project</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-13.html">Class 13 : BEDtools</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="">Class 14 : ChIP-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-15.html">Class 15 : ChIP-seq / DNase I</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-16.html">Class 16: Genomic Analysis Vignettes</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-17.html">Class 17 : Bash and Bam</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-18.html">Class 18 : R : Getting Started</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-19.html">Class 19 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-20.html">Class 20 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-21.html">Class 21 : R : DataFrames &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-22.html">Class 22 : R : Data Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-23.html">Class 23 : RNA-Seq I</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-24.html">Class 24 : RNA-Seq II</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-25.html">Class 25 : Exome-Seq - Alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-26.html">Class 26 : Exome-Seq - Variants</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../Problem_Sets.html">Problem Sets</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_1/problem_set_1.html">Problem Set 1</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_2/problem_set_2.html">Problem Set 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_3/problem_set_3.html">Problem Set 3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_4/problem_set_4.html">Problem Set 4</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_5/problem_set_5.html">Problem Set 5</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_6/problem_set_6.html">Problem Set 6</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/Problem_Set_Keys.html">Problem Set : KEYS</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../Miscellaneous.html">Miscellaneous</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../misc/commands.html">Reference: list of commands</a></li>
<li class="toctree-l2"><a class="reference internal" href="../misc/data.html">Sample Data files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../misc/tips.html">Programming Tips &amp; Tricks</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top">
        <i data-toggle="wy-nav-top" class="icon icon-reorder"></i>
        <a href="/">Genome Informatics Workshop</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <ul class="wy-breadcrumbs">
  <li><a href="../index.html">Docs</a> &raquo;</li>
  <li><a href="">Class 14 : ChIP-seq</a></li>
  
</ul>
<hr/>

          
  <div class="section" id="class-14-chip-seq">
<h1>Class 14 : ChIP-seq<a class="headerlink" href="#class-14-chip-seq" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#class-14-chip-seq" title="Slides">§</a></h1>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Class date:</th><td class="field-body">2014 Feb 28 Friday</td>
</tr>
</tbody>
</table>
<div class="section" id="goals">
<h2>Goals<a class="headerlink" href="#goals" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#goals" title="Slides">§</a></h2>
<blockquote>
<div><ol class="arabic simple">
<li>Learn the workflow for analyzing ChIP-seq data</li>
</ol>
</div></blockquote>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p><strong>LOGIN TO THE CLUSTER</strong></p>
<p class="last">You will need access to amc-tesla to do your homework. Confirm you can
log on before leaving class.</p>
</div>
</div>
<div class="section" id="chromatin-immunoprecipitation-overview">
<h2>Chromatin Immunoprecipitation Overview<a class="headerlink" href="#chromatin-immunoprecipitation-overview" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#chromatin-immunoprecipitation-overview" title="Slides">§</a></h2>
<p>Chromatin Immunoprecipitation is used to determine where a protein of
interest binds on a chromatin template <a class="reference internal" href="#park-chipseq" id="id1">[Park_Chipseq]</a>.</p>
<table class="docutils citation" frame="void" id="park-chipseq" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[Park_Chipseq]</a></td><td><a class="reference external" href="http://www.nature.com/nrg/journal/v10/n10/full/nrg2641.html">http://www.nature.com/nrg/journal/v10/n10/full/nrg2641.html</a></td></tr>
</tbody>
</table>
<img alt="../_images/chip-workflow.png" src="../_images/chip-workflow.png" />
<img alt="../_images/chip-data.png" src="../_images/chip-data.png" />
</div>
<div class="section" id="chip-seq-analysis-workflow">
<h2>ChIP-seq analysis workflow<a class="headerlink" href="#chip-seq-analysis-workflow" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#chip-seq-analysis-workflow" title="Slides">§</a></h2>
<p>A general workflow for visualizing ChIP-seq data (and many other types of
data) is:</p>
<table border="1" class="docutils">
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Operation</th>
<th class="head">File formats</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Align reads to reference genome</td>
<td><tt class="docutils literal"><span class="pre">FASTQ</span> <span class="pre">~~&gt;</span> <span class="pre">BAM</span></tt></td>
</tr>
<tr class="row-odd"><td>Generate coverage plots</td>
<td><tt class="docutils literal"><span class="pre">BAM</span> <span class="pre">~~&gt;</span> <span class="pre">bedGraph</span></tt></td>
</tr>
<tr class="row-even"><td>Call peaks</td>
<td><tt class="docutils literal"><span class="pre">BAM</span> <span class="pre">~~&gt;</span> <span class="pre">BED</span></tt></td>
</tr>
<tr class="row-odd"><td>Make binary files for UCSC display</td>
<td><tt class="docutils literal"><span class="pre">bedGraph</span> <span class="pre">~~&gt;</span> <span class="pre">bigWig</span></tt>, <tt class="docutils literal"><span class="pre">BED</span> <span class="pre">~~&gt;</span> <span class="pre">bigBed</span></tt></td>
</tr>
<tr class="row-even"><td>Identify motifs</td>
<td><tt class="docutils literal"><span class="pre">BED</span> <span class="pre">~~&gt;</span> <span class="pre">FASTA</span> <span class="pre">~~&gt;</span> <span class="pre">TXT</span> <span class="pre">/</span> <span class="pre">HTML</span></tt></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="chip-seq-data">
<h2>ChIP-seq data<a class="headerlink" href="#chip-seq-data" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#chip-seq-data" title="Slides">§</a></h2>
<p>Look at some human ChIP-seq data <a class="footnote-reference" href="#id3" id="id2">[1]</a>.</p>
<table class="docutils footnote" frame="void" id="id3" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[1]</a></td><td>Genome Browser Session
<a class="reference external" href="http://goo.gl/WfJxcM">http://goo.gl/WfJxcM</a></td></tr>
</tbody>
</table>
</div>
<div class="section" id="short-read-alignment">
<h2>Short read alignment<a class="headerlink" href="#short-read-alignment" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#short-read-alignment" title="Slides">§</a></h2>
<p>There are several short read alignment packages available. We will use
bowtie2 <a class="footnote-reference" href="#id5" id="id4">[2]</a> because it is easy to use and relatively fast.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># minimal bowtie2 command</span>
<span class="nv">$ </span>bowtie2 -x &lt;index&gt; -U &lt;read1.fq.gz&gt; <span class="o">[</span>options<span class="o">]</span> &gt; output.sam
</pre></div>
</div>
<table class="docutils footnote" frame="void" id="id5" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id4">[2]</a></td><td>Bowtie2
<a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a></td></tr>
</tbody>
</table>
<p>Normally, you would also pipe the SAM-format alignment through the samtools
suite to discard unaligned reads, sort the alignment and store it in
binary format (bam).</p>
<p>SAM format: <a class="reference external" href="http://samtools.sourceforge.net/SAMv1.pdf">http://samtools.sourceforge.net/SAMv1.pdf</a></p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># sort sam output with samtools and write bam output</span>
<span class="nv">$ </span>bowtie2 -x &lt;index&gt; -U &lt;read1.fq.gz&gt; <span class="o">[</span>options<span class="o">]</span> <span class="se">\</span>
    | samtools view -ShuF4 - <span class="se">\</span>
    | samtools sort -o - aln.temp -m 8G <span class="se">\</span>
    &gt; aln.bam
</pre></div>
</div>
</div>
<div class="section" id="generate-and-visualize-coverage-plots">
<h2>Generate and visualize coverage plots<a class="headerlink" href="#generate-and-visualize-coverage-plots" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#generate-and-visualize-coverage-plots" title="Slides">§</a></h2>
<p>Once alignment is complete, you can create coverage plots from your aligned
data, so that you can visualize your data.</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>You will need a &#8220;chromsizes&#8221; flle for many BEDtools commands. This file
contains the sizes of each chromosome in an assembly. UCSC provides a
tool to retrieve this information:</p>
<div class="last highlight-bash"><div class="highlight"><pre><span class="c"># retrieve chrom sizes for the hg19 assembly and write them to a file</span>
<span class="c"># inspect this file so you know what it looks like</span>
<span class="nv">$ </span>fetchChromSizes hg19 &gt; hg19.chrom.sizes
</pre></div>
</div>
</div>
<div class="section" id="coverage-plots-with-bedtools">
<h3>Coverage plots with BEDtools<a class="headerlink" href="#coverage-plots-with-bedtools" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#coverage-plots-with-bedtools" title="Slides">§</a></h3>
<p>To generate coverage plots, we will use BEDtools. Here, we&#8217;ll
use the <a class="reference external" href="http://bedtools.readthedocs.org/en/latest/content/tools/genomecov.html#genomecov" title="(in bedtools v2.20.1)"><em class="xref std std-ref">genomecov</em></a> tool.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># -bg : write ouptput in bedGraph format</span>
<span class="nv">$ </span>bedtools genomecov -i &lt;aln.bam&gt; -g &lt;chrom.sizes&gt; -bg &gt; coverage.bg
</pre></div>
</div>
<p>This command writes a bedGraph format file called <tt class="docutils literal"><span class="pre">coverage.bg</span></tt>. Use
<tt class="docutils literal"><span class="pre">less</span></tt> to examine this file.</p>
<p>Words to live by: <strong>If you make a BED file, sort the BED file</strong></p>
<p>Many strange things can happen if you use unsorted BED files for
analysis. Once you create a BED file, sort it with one of these:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># using UNIX sort</span>
<span class="nv">$ </span>sort -k1,1 -k2,2 unsorted.bed &gt; sorted.bed

<span class="c"># UCSC tool. same filename twice, overwrites original file</span>
<span class="nv">$ </span>bedSort file.bed file.bed

<span class="c"># or you can use bedtools; writes additional file</span>
<span class="nv">$ </span>bedtools sort -i - &lt; unsorted.bed &gt; sorted.bed
</pre></div>
</div>
</div>
<div class="section" id="coverage-plots-split-by-strand">
<h3>Coverage plots split by strand<a class="headerlink" href="#coverage-plots-split-by-strand" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#coverage-plots-split-by-strand" title="Slides">§</a></h3>
<p>For some experiments, you will analyze the data relative to each strand of
the reference genome. For example, RNA is transcribed in single-stranded
form and derives from one or the other strand.</p>
<p>During alignment, reads from an RNA-based experiment will map to either
the positive (&#8216;+&#8217; or <tt class="docutils literal"><span class="pre">pos</span></tt>) or negative (&#8216;-&#8216; or <tt class="docutils literal"><span class="pre">neg</span></tt>) strand. You can
generate signal plots for <tt class="docutils literal"><span class="pre">pos</span></tt> and <tt class="docutils literal"><span class="pre">neg</span></tt> strands separately with
<tt class="docutils literal"><span class="pre">bedtools</span></tt>:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ common_args</span><span class="o">=</span><span class="s2">&quot;-ibam &lt;aln.bam&gt; -g &lt;chrom.size&gt; -bg&quot;</span>
<span class="nv">$ </span>bedtools genomecov <span class="nv">$common_args</span> -strand + &gt; coverage.pos.bg
<span class="nv">$ </span>bedtools genomecov <span class="nv">$common_args</span> -strand - &gt; coverage.neg.bg
</pre></div>
</div>
<p>You would then create bigWigs for each of these display the stranded data
in the Genome Browser.</p>
</div>
</div>
<div class="section" id="peak-calling">
<h2>Peak calling<a class="headerlink" href="#peak-calling" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#peak-calling" title="Slides">§</a></h2>
<p>There are several available software packages for identifying regions
enriched in your IP experiment (i.e. peaks). We will use macs2 here.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># minimal macs2 command</span>
<span class="nv">$ </span>macs2 callpeak --treatment &lt;aln.bam&gt; --name &lt;exp.name&gt; <span class="o">[</span>options<span class="o">]</span>
</pre></div>
</div>
</div>
<div class="section" id="identify-sequence-motifs-in-enriched-regions">
<h2>Identify sequence motifs in enriched regions<a class="headerlink" href="#identify-sequence-motifs-in-enriched-regions" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#identify-sequence-motifs-in-enriched-regions" title="Slides">§</a></h2>
<p>You can use meme <a class="footnote-reference" href="#id7" id="id6">[3]</a> to identify over-represented motifs in groups of
sequences (e.g. sequences covered by ChIP peaks).</p>
<p>Use the <a class="reference external" href="http://bedtools.readthedocs.org/en/latest/content/tools/getfasta.html#getfasta" title="(in bedtools v2.20.1)"><em class="xref std std-ref">bedtools getfasta</em></a> command to fetch
fasta sequences.</p>
<p>Note: meme looks at both strands of a DNA sequence by default.</p>
<table class="docutils footnote" frame="void" id="id7" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id6">[3]</a></td><td>MEME
<a class="reference external" href="http://meme.nbcr.net/meme/">http://meme.nbcr.net/meme/</a></td></tr>
</tbody>
</table>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>bedtools getfasta -fi &lt;ref.fa&gt; -bed &lt;peaks.bed&gt; -fo peaks.fa
<span class="nv">$ </span>meme -nmotifs 5 -minw 6 -maxw 20 -dna &lt;peaks.fa&gt;
</pre></div>
</div>
</div>
<div class="section" id="putting-it-all-together">
<h2>Putting it all together<a class="headerlink" href="#putting-it-all-together" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-14.html#putting-it-all-together" title="Slides">§</a></h2>
<p>Here is a script that combines the above in a single workflow:</p>
<div class="highlight-bash"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79</pre></div></td><td class="code"><div class="highlight"><pre><span class="c">#! /usr/bin/env bash</span>

<span class="c">#BSUB -J chipseq</span>
<span class="c">#BSUB -o %J.out</span>
<span class="c">#BSUB -e %J.err</span>

<span class="c">#</span>
<span class="c"># load on the head-node required modules before submitting this script</span>
<span class="c">#</span>
<span class="c"># $ module load modules modules-init modules-python</span>
<span class="c"># $ module load python</span>
<span class="c"># $ module load bowtie2</span>
<span class="c"># $ module load ucsc</span>
<span class="c"># $ module load bedtools</span>
<span class="c"># $ module load meme</span>
<span class="c">#</span>

<span class="nb">set</span> -e

<span class="nv">data</span><span class="o">=</span>/vol1/opt/data
<span class="nv">fasta</span><span class="o">=</span><span class="nv">$data</span>/hg19.fa
<span class="nv">bwtindex</span><span class="o">=</span><span class="nv">$data</span>/hg19
<span class="nv">chromsize</span><span class="o">=</span><span class="nv">$data</span>/hg19.chrom.sizes

<span class="c"># downloaded from ENCODE website - H3K4me3 ChIP in Hela cells</span>
<span class="nv">fastqfile</span><span class="o">=</span><span class="nv">$data</span>/wgEncodeBroadHistoneHelas3H3k4me3StdRawDataRep1.fastq.gz

<span class="nv">expname</span><span class="o">=</span><span class="s2">&quot;chipseq.hela.h3k4me3&quot;</span>
<span class="nv">bamfile</span><span class="o">=</span><span class="nv">$expname</span>.bam

<span class="c"># run the alignment</span>
bowtie2 -x <span class="nv">$bwtindex</span> -U <span class="nv">$fastqfile</span> <span class="se">\</span>
    | samtools view -ShuF4 - <span class="se">\</span>
    | samtools sort -o - aln.temp -m 8G <span class="se">\</span>
    &gt; <span class="nv">$bamfile</span>

<span class="c"># call peaks</span>
macs2 callpeak --treatment <span class="nv">$bamfile</span> --name <span class="nv">$expname</span>

<span class="c"># macs2 will generate a series of files, including one in</span>
<span class="c"># ``narrowPeak`` format, an extended BED fromat:</span>
<span class="nv">peakbed</span><span class="o">=</span>chipseq.hela.h3k4me3_peaks.narrowPeak
<span class="nv">peakbigbed</span><span class="o">=</span>chipseq.hela.h3k4me3_peaks.bb
bedToBigBed <span class="nv">$peakbed</span> <span class="nv">$chromsize</span> <span class="nv">$peakbigbed</span>

<span class="c"># make coverage plots in bedgraph and bigwig</span>
<span class="nv">bedgraph</span><span class="o">=</span><span class="s2">&quot;$expname.bg&quot;</span>
<span class="nv">bigwig</span><span class="o">=</span><span class="s2">&quot;$expname.bw&quot;</span>
bedtools genomecov -ibam <span class="nv">$bamfile</span> -g <span class="nv">$chromsize</span> -bg &gt; <span class="nv">$bedgraph</span>
bedGraphToBigWig <span class="nv">$bedgraph</span> <span class="nv">$chromsize</span> <span class="nv">$bigwig</span>

<span class="c"># find motifs on chr22</span>
<span class="nv">peakfasta</span><span class="o">=</span><span class="nv">$expname</span>.peaks.fa
<span class="nv">peak_chr22_fasta</span><span class="o">=</span><span class="nv">$expname</span>.peaks.chr22.fa
bedtools getfasta -fi <span class="nv">$fasta</span> -bed <span class="nv">$peakbed</span> -fo <span class="nv">$peakfasta</span>
awk <span class="s1">&#39;$1 == &quot;chr22&quot;&#39;</span> &lt; <span class="nv">$peakfasta</span> &gt; <span class="nv">$peak_chr22_fasta</span>

<span class="c"># XXX this will take a while to run ...</span>
<span class="nv">meme_args</span><span class="o">=</span><span class="s2">&quot;-nmotifs 5 -minw 6 -maxw 20 -maxsize 10000000 -dna&quot;</span>
meme <span class="nv">$meme_args</span> <span class="nv">$peak_chr22_fasta</span>

<span class="c"># cp the bigWig / bigBed to public_html directory</span>
<span class="c">#</span>
<span class="c"># bigWig trackline (needs to be all on one line)</span>
<span class="c">#</span>
<span class="c"># track type=bigWig</span>
<span class="c"># name=&#39;coverage&#39;</span>
<span class="c"># bigDataUrl=&quot;http://amc-sandbox.ucdenver.edu/~username/file.bw</span>
<span class="c"># visibility=full</span>
<span class="c"># maxHeightPixels=50:35:35</span>
<span class="c">#</span>
<span class="c"># bigBed trackline</span>
<span class="c">#</span>
<span class="c"># track type=bigBed</span>
<span class="c"># name=&#39;peaks&#39;</span>
<span class="c"># bigDataUrl=&quot;http://amc-sandbox.ucdenver.edu/~username/file.bb</span>
<span class="c"># visibility=dense</span>
<span class="c"># maxHeightPixels=50:35:35</span>
<span class="c">#</span>
</pre></div>
</td></tr></table></div>
</div>
</div>


          <footer>
  
    <div class="rst-footer-buttons">
      
        <a href="class-15.html" class="btn btn-neutral float-right" title="Class 15 : ChIP-seq / DNase I"/>Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="class-13.html" class="btn btn-neutral" title="Class 13 : BEDtools"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <p>
      &copy; Copyright 2014, Brent Pedersen, Jay Hesselberth.
    Last updated on Aug 25, 2014.
  </p>

  <a href="https://www.github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="http://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  

</body>
</html>