

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Class 23 : RNA-Seq I &mdash; Genome Informatics Workshop 0.1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  

    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:'../',
        VERSION:'0.1',
        COLLAPSE_INDEX:false,
        FILE_SUFFIX:'.html',
        HAS_SOURCE:  true
      };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
    <link rel="top" title="Genome Informatics Workshop 0.1 documentation" href="../index.html"/>
        <link rel="up" title="Class list" href="../Classes.html"/>
        <link rel="next" title="Class 24 : RNA-Seq II" href="class-24.html"/>
        <link rel="prev" title="Class 22 : R : Data Manipulation &amp; Plotting" href="class-22.html"/> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>



  <script src="//cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="../index.html" class="icon icon-home"> Genome Informatics Workshop</a>
        <form class="wy-form" action="../search.html" method="get">
  <input type="text" name="q" placeholder="Search docs" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../Syllabus.html">Syllabus (MOLB FIXME)</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal" href="../Classes.html">Class list</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="class-1.html">Class 1 : Virtual Machine (VM) and Linux Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-2.html">Class 2 : The command-line</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-3.html">Class 3 : The command-line (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-4.html">Class 4 : awk</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-5.html">Class 5 : Working in a cluster environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-6.html">Class 6 : Working in a cluster environment (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-7.html">Class 7 : Python : Basics</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-8.html">Class 8 : Python : Basics (2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-9.html">Class 9 : Python Concepts</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-10.html">Class 10 : Applied Python</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-11.html">Class 11 : Python Idioms</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-12.html">Class 12 : The ENCODE Project</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-13.html">Class 13 : BEDtools</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-14.html">Class 14 : ChIP-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-15.html">Class 15 : ChIP-seq / DNase I</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-16.html">Class 16: Genomic Analysis Vignettes</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-17.html">Class 17 : Bash and Bam</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-18.html">Class 18 : R : Getting Started</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-19.html">Class 19 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-20.html">Class 20 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-21.html">Class 21 : R : DataFrames &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-22.html">Class 22 : R : Data Manipulation &amp; Plotting</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="">Class 23 : RNA-Seq I</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-24.html">Class 24 : RNA-Seq II</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-25.html">Class 25 : Exome-Seq - Alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-26.html">Class 26 : Exome-Seq - Variants</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../Problem_Sets.html">Problem Sets</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_1/problem_set_1.html">Problem Set 1</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_2/problem_set_2.html">Problem Set 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_3/problem_set_3.html">Problem Set 3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_4/problem_set_4.html">Problem Set 4</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_5/problem_set_5.html">Problem Set 5</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_6/problem_set_6.html">Problem Set 6</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/Problem_Set_Keys.html">Problem Set : KEYS</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../Miscellaneous.html">Miscellaneous</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../misc/commands.html">Reference: list of commands</a></li>
<li class="toctree-l2"><a class="reference internal" href="../misc/data.html">Sample Data files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../misc/tips.html">Programming Tips &amp; Tricks</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top">
        <i data-toggle="wy-nav-top" class="icon icon-reorder"></i>
        <a href="/">Genome Informatics Workshop</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <ul class="wy-breadcrumbs">
  <li><a href="../index.html">Docs</a> &raquo;</li>
  <li><a href="">Class 23 : RNA-Seq I</a></li>
  
</ul>
<hr/>

          
  <div class="section" id="class-23-rna-seq-i">
<h1>Class 23 : RNA-Seq I<a class="headerlink" href="#class-23-rna-seq-i" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#class-23-rna-seq-i" title="Slides">§</a></h1>
<div class="section" id="goals">
<h2>Goals<a class="headerlink" href="#goals" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#goals" title="Slides">§</a></h2>
<blockquote>
<div><ol class="arabic simple">
<li>Why RNA-Seq</li>
<li>QC.</li>
<li>Align Reads</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="rna-seq">
<h2>RNA-Seq<a class="headerlink" href="#rna-seq" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#rna-seq" title="Slides">§</a></h2>
<blockquote>
<div><ul class="simple">
<li>We use RNA-Seq primarily to measure gene expression</li>
<li>We can also find transcript use (transcript-switching/alternative-splicing)</li>
<li>Fusion Transcripts</li>
<li>Possible to call variants (SNPs) on RNA-Seq data</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="library-prep">
<h2>Library Prep<a class="headerlink" href="#library-prep" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#library-prep" title="Slides">§</a></h2>
<p>Libraries were prepared by a standard RNA-seq protocol including:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Oligo-dT purification of polyadenylated mRNA</li>
<li>Reverse transcription with random nonamers</li>
<li>Second strand sythesis to create double stranded DNA (dsDNA)</li>
<li>Shearing of dsDNA to 300-600 bp</li>
<li>End polishing, adaptor ligation, PCR with indexed primers</li>
<li>Quantify libraries, Mix, sequence.</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="example-data">
<h2>Example Data<a class="headerlink" href="#example-data" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#example-data" title="Slides">§</a></h2>
<p>We will look at ~10 million 75 base paired-end reads from 4 RNA-seq
libraries prepared from these <cite>C. elegans</cite> strains:</p>
<table border="1" class="docutils">
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Strain</th>
<th class="head">Treatment</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>RtcB wild-type</td>
<td>No treatment</td>
</tr>
<tr class="row-odd"><td>RtcB wild-type</td>
<td>Tunicamycin (induces UPR)</td>
</tr>
<tr class="row-even"><td>RtcB mutant</td>
<td>No treatment</td>
</tr>
<tr class="row-odd"><td>RtcB mutant</td>
<td>Tunicamycin (induces UPR)</td>
</tr>
</tbody>
</table>
<p>RtcB is an RNA ligase. RtcB wild-type and mutant strains were treated with
tunicamycin, which inhibits protein glycosylation in the endoplasmic
reticulum and triggers the &#8220;unfolded protein reponse&#8221;:</p>
<div class="highlight-python"><div class="highlight"><pre>ER stress ~~&gt;
    Ire1 multimerization ~~&gt;
        Ire1 ribonuclease domain cuts
        xbp-1 transcript to release intron ~~&gt;
            RtcB ligates xbp-1 exons together ~~&gt;
                Xbp-1 protein is made, which activates UPR-related genes
                in the nucleus.
</pre></div>
</div>
<p>The expectation is that RtcB mutants will be unable to ligate specific RNA
substrates together, thus the mRNA library in the mutant might contain
altered levels of RNA fragments relative to the wild-type. In particular,
xbp-1 <a class="footnote-reference" href="#id2" id="id1">[1]</a> fragments should accumulate in the mutant.</p>
<table class="docutils footnote" frame="void" id="id2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[1]</a></td><td><a class="reference external" href="http://www.wormbase.org/species/c_elegans/gene/WBGene00006959?query=xbp-1#0-9e-3">http://www.wormbase.org/species/c_elegans/gene/WBGene00006959?query=xbp-1#0-9e-3</a></td></tr>
</tbody>
</table>
</div>
<div class="section" id="experimental-design">
<h2>Experimental Design<a class="headerlink" href="#experimental-design" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#experimental-design" title="Slides">§</a></h2>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p>Please do not do RNA-Seq on fewer than 6 samples (3 vs 3).</p>
<p class="last">Even in a pilot study.</p>
</div>
<p>See: <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/30/3/301">http://bioinformatics.oxfordjournals.org/content/30/3/301</a></p>
<p>Titled: <strong>RNA-seq differential expression studies: more sequence or more
replication?</strong></p>
<p>Answer: <strong>more replication</strong></p>
<p>Your bioinformatician will thank you.</p>
</div>
<div class="section" id="analysis-plan">
<h2>Analysis Plan<a class="headerlink" href="#analysis-plan" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#analysis-plan" title="Slides">§</a></h2>
<blockquote>
<div><ol class="arabic simple">
<li>QC reads with Fastqc</li>
<li>Trim reads by quality with sickle</li>
<li>Re-check QC on trimmed reads</li>
<li>Align reads with Tophat<ol class="arabic">
<li>give a GTF to help it to find known transcripts</li>
<li>also allow novel junctions</li>
<li>fusion transcripts</li>
</ol>
</li>
<li>Count reads by gene with featureCounts (from subread)</li>
<li>Differential Expression statistics with DESeq2</li>
<li>Differential Transcript Use and fusion transcripts (if time)</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="qc-reads">
<h2>QC Reads<a class="headerlink" href="#qc-reads" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#qc-reads" title="Slides">§</a></h2>
<p>We will use fastqc to check the quality of the reads and look for adapter
contamination</p>
<blockquote>
<div>$ fastqc -o ~/public_html/fqc-$sample $data/${sample}_R2.fastq.gz</div></blockquote>
</div>
<div class="section" id="quality-trim-reads">
<h2>Quality Trim Reads<a class="headerlink" href="#quality-trim-reads" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#quality-trim-reads" title="Slides">§</a></h2>
<p>We will trim low-quality bases from reads using sickle:
<a class="reference external" href="https://github.com/najoshi/sickle">https://github.com/najoshi/sickle</a></p>
<p>Lenient trimming has been shown to improve mapping rates in
RNA-Seq. See, e.g. <a class="reference external" href="http://genomebio.org/is-trimming-is-beneficial-in-rna-seq/">http://genomebio.org/is-trimming-is-beneficial-in-rna-seq/</a></p>
</div>
<div class="section" id="qc-quality-trimmed-reads">
<h2>QC Quality Trimmed Reads<a class="headerlink" href="#qc-quality-trimmed-reads" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#qc-quality-trimmed-reads" title="Slides">§</a></h2>
<p>Compare before/after. Makes most difference in bad datasets.</p>
</div>
<div class="section" id="align-with-tophat2">
<h2>Align with Tophat2<a class="headerlink" href="#align-with-tophat2" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#align-with-tophat2" title="Slides">§</a></h2>
<p>Many RNA-Seq (splice-aware) aligners:</p>
<blockquote>
<div><ul class="simple">
<li>Tophat/Tophat2</li>
<li>GSNAP</li>
<li>STAR</li>
<li>RUM</li>
<li>MapSplice</li>
<li>etc.</li>
</ul>
</div></blockquote>
<p>We will use tophat2.</p>
</div>
<div class="section" id="igenomes">
<h2>IGenomes<a class="headerlink" href="#igenomes" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#igenomes" title="Slides">§</a></h2>
<p>If you work with a model organism, you can get bowtie (tophat2) genome
indexes and feature annotation in normalized format from Illumina.</p>
<p><a class="reference external" href="http://tophat.cbcb.umd.edu/igenomes.shtml">http://tophat.cbcb.umd.edu/igenomes.shtml</a></p>
<p>We will use the <em>C. elegans</em> data downloaded to <cite>amc-tesla:~brentp/data/ce/</cite></p>
<blockquote>
<div><ul class="simple">
<li>Using this will save you a lot of trouble</li>
<li>Tophat2 uses known transcripts and attempts to align to those as well as to
novel transcripts</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="spliced-alignment">
<h2>Spliced Alignment<a class="headerlink" href="#spliced-alignment" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#spliced-alignment" title="Slides">§</a></h2>
<p>From tophat paper:</p>
<img alt="http://bioinformatics.oxfordjournals.org/content/25/9/1105/F1.large.jpg" src="http://bioinformatics.oxfordjournals.org/content/25/9/1105/F1.large.jpg" />
</div>
<div class="section" id="id3">
<h2>Spliced Alignment<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#id3" title="Slides">§</a></h2>
<p>From tophat2 paper:</p>
<img alt="http://genomebiology.com/content/figures/gb-2013-14-4-r36-1-l.jpg" src="http://genomebiology.com/content/figures/gb-2013-14-4-r36-1-l.jpg" />
</div>
<div class="section" id="tophat2-invocation">
<h2>Tophat2 Invocation<a class="headerlink" href="#tophat2-invocation" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#tophat2-invocation" title="Slides">§</a></h2>
<div class="highlight-bash"><div class="highlight"><pre>tophat2 -o <span class="nv">$out</span>/results/<span class="nv">$sample</span> <span class="nv">$reference</span> <span class="nv">$fq_1</span> <span class="nv">$fq2</span> <span class="se">\</span>
    --fusion-search -p 6 --transcriptome_index <span class="nv">$TINDEX</span> <span class="se">\</span>
    --GTF <span class="nv">$ANNOTATION_GTF</span>
</pre></div>
</div>
<p>Output will be in accepted_hits.bam</p>
</div>
<div class="section" id="just-do-it">
<h2>Just Do It<a class="headerlink" href="#just-do-it" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-23.html#just-do-it" title="Slides">§</a></h2>
<p>On amc-tesla</p>
<div class="highlight-bash"><div class="highlight"><pre>mkdir -p ~/class-23/src/
cp ~brentp/class-23/run.sh ~/class-23/
cp ~brentp/class-23/src/clean-counts.py ~/class-23/src/
<span class="nb">cd</span> ~/class-23/src/
module load sickle
module load subread
module load bowtie2
</pre></div>
</div>
</div>
</div>


          <footer>
  
    <div class="rst-footer-buttons">
      
        <a href="class-24.html" class="btn btn-neutral float-right" title="Class 24 : RNA-Seq II"/>Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="class-22.html" class="btn btn-neutral" title="Class 22 : R : Data Manipulation &amp; Plotting"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <p>
      &copy; Copyright 2014, Brent Pedersen, Jay Hesselberth.
    Last updated on Aug 25, 2014.
  </p>

  <a href="https://www.github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="http://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  

</body>
</html>