<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Class 26 : Exome-Seq - Variants &mdash; Genome Informatics Workshop 0.2 documentation</title>
    
    <link rel="stylesheet" href="../_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/bootswatch-3.1.0/flatly/bootstrap.min.css" type="text/css" />
    <link rel="stylesheet" href="../_static/bootstrap-sphinx.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="../_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="../_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="../_static/bootstrap-3.1.0/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="../_static/bootstrap-sphinx.js"></script>
    <link rel="top" title="Genome Informatics Workshop 0.2 documentation" href="../index.html" />
    <link rel="up" title="Class list" href="../Classes.html" />
    <link rel="next" title="Problem Sets" href="../Problem_Sets.html" />
    <link rel="prev" title="Class 25 : Exome-Seq - Alignment" href="class-25.html" /> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>


  </head>
  <body>

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../index.html"><img src="../_static/my_logo.png">
          Genome Informatics Workshop</a>
        <span class="navbar-text navbar-version pull-left"><b>0.2</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            <li class="divider-vertical"></li>
            
                <li><a href="http://hesselberthlab.github.io">Hesselberth lab</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="../index.html">Genome Informatics Workshop <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../Syllabus.html">Syllabus</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../Classes.html">Class list</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Problem_Sets.html">Problem Sets</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Miscellaneous.html">Miscellaneous</a></li>
</ul>
</ul>
</li>
              
            
            
            
            
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
      <div class="col-md-3">
        <div id="sidebar" class="bs-sidenav" role="complementary"><ul>
<li><a class="reference internal" href="#">Class 26 : Exome-Seq - Variants</a><ul>
<li><a class="reference internal" href="#goals">Goals</a></li>
<li><a class="reference internal" href="#lsf-job-arrays">LSF Job Arrays</a></li>
<li><a class="reference internal" href="#lsf-job-arrays-ii">LSF Job Arrays II</a></li>
<li><a class="reference internal" href="#exercise">Exercise</a></li>
<li><a class="reference internal" href="#bams">BAMs</a></li>
<li><a class="reference internal" href="#variant-call-format-vcf">Variant Call Format (VCF)</a></li>
<li><a class="reference internal" href="#project-design">Project Design</a></li>
<li><a class="reference internal" href="#snpeff-snpsift">SNPEff/SNPSift</a></li>
<li><a class="reference internal" href="#java">Java</a></li>
<li><a class="reference internal" href="#filters">Filters</a></li>
<li><a class="reference internal" href="#filters-ii">Filters II</a></li>
<li><a class="reference internal" href="#filters-iii">Filters III</a></li>
<li><a class="reference internal" href="#filters-applied">Filters Applied</a></li>
<li><a class="reference internal" href="#filtered">Filtered</a></li>
<li><a class="reference internal" href="#viewing-on-ucsc">Viewing on UCSC</a></li>
<li><a class="reference internal" href="#viewing-in-igv">Viewing in IGV</a></li>
<li><a class="reference internal" href="#viewing-with-samtools-tview">Viewing with samtools tview</a></li>
<li><a class="reference internal" href="#extracting-parts-of-the-alignment">Extracting parts of the alignment</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    <div class="col-md-9">
      
  <div class="section" id="class-26-exome-seq-variants">
<h1>Class 26 : Exome-Seq - Variants<a class="headerlink" href="#class-26-exome-seq-variants" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#class-26-exome-seq-variants" title="Slides">§</a></h1>
<div class="section" id="goals">
<h2>Goals<a class="headerlink" href="#goals" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#goals" title="Slides">§</a></h2>
<blockquote>
<div><ol class="arabic simple">
<li>Review cluster usage</li>
<li>BAM format</li>
<li>VCF format</li>
<li>Filter VCF</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="lsf-job-arrays">
<h2>LSF Job Arrays<a class="headerlink" href="#lsf-job-arrays" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#lsf-job-arrays" title="Slides">§</a></h2>
<p>Run a command (<cite>bwa mem</cite>) on 8 samples</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c">#BSUB -J align[1-8]</span>

<span class="nv">samples</span><span class="o">=(</span>ll-1 ll-3 ll-5 ll-7 lll-10
         lll-1 lll-2 lll-9<span class="o">)</span>
<span class="nv">sample</span><span class="o">=</span><span class="k">${</span><span class="nv">samples</span><span class="p">[</span><span class="k">$((</span><span class="nv">$LSB_JOBINDEX</span> <span class="o">-</span> <span class="m">1</span><span class="k">))</span><span class="p">]</span><span class="k">}</span>

<span class="nv">R1</span><span class="o">=</span><span class="nv">$FQDIR</span>/<span class="k">${</span><span class="nv">sample</span><span class="k">}</span>_R1_001.fastq.gz
<span class="nv">R2</span><span class="o">=</span><span class="nv">$FQDIR</span>/<span class="k">${</span><span class="nv">sample</span><span class="k">}</span>_R2_001.fastq.gz

bwa mem hg19.fa <span class="nv">$R1</span> <span class="nv">$R2</span> &gt; <span class="nv">$sample</span>.sam
</pre></div>
</div>
</div>
<div class="section" id="lsf-job-arrays-ii">
<h2>LSF Job Arrays II<a class="headerlink" href="#lsf-job-arrays-ii" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#lsf-job-arrays-ii" title="Slides">§</a></h2>
<p>Same pipeline on different data (16 samples)</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c">#BSUB -J align[1-16]</span>

<span class="nv">samples</span><span class="o">=(</span>1_Y_Spleen_10_28_ATCACG 2_Y_10_31_CGATGT 3_Y2_11_25_TTAGGC
4_M_Spleen_10_28_TGACCA .. <span class="o">[</span>16 total samples<span class="o">]</span> ..<span class="o">)</span>

<span class="nv">sample</span><span class="o">=</span><span class="k">${</span><span class="nv">samples</span><span class="p">[</span><span class="k">$((</span><span class="nv">$LSB_JOBINDEX</span> <span class="o">-</span> <span class="m">1</span><span class="k">))</span><span class="p">]</span><span class="k">}</span>

<span class="nv">R1</span><span class="o">=</span><span class="nv">$FQDIR</span>/<span class="k">${</span><span class="nv">sample</span><span class="k">}</span>_L008_R1_001.fastq.gz
<span class="nv">R2</span><span class="o">=</span><span class="nv">$FQDIR</span>/<span class="k">${</span><span class="nv">sample</span><span class="k">}</span>_L008_R2_001.fastq.gz

bwa mem mm10.fa <span class="nv">$R1</span> <span class="nv">$R2</span> &gt; <span class="nv">$sample</span>.sam
</pre></div>
</div>
</div>
<div class="section" id="exercise">
<h2>Exercise<a class="headerlink" href="#exercise" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#exercise" title="Slides">§</a></h2>
<p>use a job array to submit a script 10 times where
each on echos it&#8217;s number in the array:</p>
<div class="highlight-python"><div class="highlight"><pre>hello from 1
hello from 2
...
hello from 10
</pre></div>
</div>
<p>specify <cite>#BSUB -e hello.%I.%J.err</cite> and <cite>#BSUB -o hello.%I.%J.out</cite></p>
<p>usually good to debug with a single job:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="c">#BSUB -J hello[1]</span>
</pre></div>
</div>
<p>then expand to number of jobs once that works.</p>
</div>
<div class="section" id="bams">
<h2>BAMs<a class="headerlink" href="#bams" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#bams" title="Slides">§</a></h2>
<p>BAMs are the alignment format. Here is one where we&#8217;ve marked
duplicates.</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools view -h ~brentp/exomes/results/ll-7.dedup.bam | less
</pre></div>
</div>
</div>
<div class="section" id="variant-call-format-vcf">
<h2>Variant Call Format (VCF)<a class="headerlink" href="#variant-call-format-vcf" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#variant-call-format-vcf" title="Slides">§</a></h2>
<p><a class="reference external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41">http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41</a></p>
<p>Our VCFs contain the 8 samples in columns 10+. They are split by chromosome.</p>
<p>We want to filter our VCF according to our study design.</p>
</div>
<div class="section" id="project-design">
<h2>Project Design<a class="headerlink" href="#project-design" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#project-design" title="Slides">§</a></h2>
<img alt="../_images/phenotype.png" src="../_images/phenotype.png" />
</div>
<div class="section" id="snpeff-snpsift">
<h2>SNPEff/SNPSift<a class="headerlink" href="#snpeff-snpsift" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#snpeff-snpsift" title="Slides">§</a></h2>
<blockquote>
<div><ul class="simple">
<li>find <strong>Eff</strong> ect of variant (missense/nonsense/etc)</li>
<li>filter variants by presence in DBSNP</li>
<li>filter variants by genotype across samples</li>
<li>filter variants by <em>lots of stuff</em></li>
</ul>
</div></blockquote>
<p><a class="reference external" href="http://snpeff.sourceforge.net/SnpEff_manual.html">http://snpeff.sourceforge.net/SnpEff_manual.html</a></p>
<p>Remember we annotated with DBSNP and by effect.</p>
</div>
<div class="section" id="java">
<h2>Java<a class="headerlink" href="#java" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#java" title="Slides">§</a></h2>
<div class="highlight-bash"><div class="highlight"><pre>module load java/1.7
qlogin
java -Xmx4G -jar ~brentp/opt/snpEff/SnpSift.jar
</pre></div>
</div>
</div>
<div class="section" id="filters">
<h2>Filters<a class="headerlink" href="#filters" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#filters" title="Slides">§</a></h2>
<blockquote>
<div><ul class="simple">
<li>Get variants with quality score greater than 40:</li>
</ul>
</div></blockquote>
<div class="highlight-bash"><div class="highlight"><pre>cat  ~brentp/exomes/results/vcfs/chr18.snpeff.dbsnp.vcf | <span class="se">\</span>
java -Xmx4G -jar ~brentp/opt/snpEff/SnpSift.jar <span class="se">\</span>
 filter  <span class="s2">&quot;((QUAL &gt;= 40))&quot;</span> | less
</pre></div>
</div>
</div>
<div class="section" id="filters-ii">
<h2>Filters II<a class="headerlink" href="#filters-ii" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#filters-ii" title="Slides">§</a></h2>
<p>To (kinda) match the study design. Get variants with:</p>
<blockquote>
<div><ul class="simple">
<li>quality score greater than 40</li>
<li>with 5 hets and 3 reference</li>
<li>high/moderate impact</li>
</ul>
</div></blockquote>
<div class="highlight-bash"><div class="highlight"><pre>cat  ~brentp/exomes/results/vcfs/chr18.snpeff.dbsnp.vcf | <span class="se">\</span>
java -Xmx4G -jar ~brentp/opt/snpEff/SnpSift.jar <span class="se">\</span>
 filter  <span class="s2">&quot;(</span>
<span class="s2"> (QUAL &gt;= 40) &amp;</span>
<span class="s2"> (countHet() = 5 &amp; countRef() = 3) &amp;</span>
<span class="s2"> ((EFF[*].IMPACT = &#39;HIGH&#39;) | (EFF[*].IMPACT = &#39;MODERATE&#39;))</span>
<span class="s2"> ) &quot;</span> | less
</pre></div>
</div>
<p>NOTE extra parens around impact of <cite>HIGH</cite> or <cite>MODERATE</cite></p>
<p>How many variants does that give us?</p>
</div>
<div class="section" id="filters-iii">
<h2>Filters III<a class="headerlink" href="#filters-iii" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#filters-iii" title="Slides">§</a></h2>
<p>Many more filtering options here:
<a class="reference external" href="http://snpeff.sourceforge.net/SnpSift.html">http://snpeff.sourceforge.net/SnpSift.html</a></p>
<p>For example: check if the first sample&#8217;s genotype is a variant
and the 2nd samples is reference</p>
<blockquote>
<div>&#8220;isVariant(GEN[0]) &amp; isRef(GEN[1])&#8221;</div></blockquote>
<p><em>use this to extract variants that match our exact criteria</em></p>
</div>
<div class="section" id="filters-applied">
<h2>Filters Applied<a class="headerlink" href="#filters-applied" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#filters-applied" title="Slides">§</a></h2>
<p>Look at the header in a VCF:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>grep -m 1 <span class="s2">&quot;#CHROM&quot;</span> ~brentp/exomes/results/vcfs/chr18.snpeff.dbsnp.vcf
<span class="c">#CHROM  POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  ll-1    ll-3    ll-5</span>
ll-7    lll-1   lll-10  lll-2   lll-9
</pre></div>
</div>
<p>Match that sample order with the pedigree image to get a filter like:</p>
<div class="highlight-python"><div class="highlight"><pre>isVariant(GEN[0]) &amp; isVariant(GEN[3]) \
    &amp; isVariant(GEN[4]) ...
</pre></div>
</div>
</div>
<div class="section" id="filtered">
<h2>Filtered<a class="headerlink" href="#filtered" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#filtered" title="Slides">§</a></h2>
<div class="highlight-bash"><div class="highlight"><pre>cat  ~brentp/exomes/results/vcfs/chr*.snpeff.dbsnp.vcf | <span class="se">\</span>
java -Xmx4G -jar ~brentp/opt/snpEff/SnpSift.jar <span class="se">\</span>
    filter  <span class="se">\</span>
    <span class="s2">&quot;((QUAL &gt;= 40) &amp; isVariant(GEN[0]) &amp; isVariant(GEN[3]) \</span>
<span class="s2">    &amp; isVariant(GEN[4]) &amp; isVariant(GEN[7]) &amp; isVariant(GEN[5]) \</span>
<span class="s2">    &amp; isRef(GEN[1]) &amp; isRef(GEN[2]) &amp; isRef(GEN[6]) \</span>
<span class="s2">    &amp; (DP &gt;= 40) \</span>
<span class="s2">    &amp; (( na FILTER) | (FILTER = &#39;PASS&#39;)) \</span>
<span class="s2">    &amp; ((EFF[*].IMPACT = &#39;HIGH&#39;) | (EFF[*].IMPACT = &#39;MODERATE&#39;))</span>
<span class="s2">        )&quot;</span>
</pre></div>
</div>
<p>Since we are likely interested in novel variants, we can then filter
to exclude things with an <cite>rs</cite> number.</p>
</div>
<div class="section" id="viewing-on-ucsc">
<h2>Viewing on UCSC<a class="headerlink" href="#viewing-on-ucsc" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#viewing-on-ucsc" title="Slides">§</a></h2>
<blockquote>
<div><a class="reference external" href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;position=chr12%3A12022530-12022539&amp;hg.customText=http://amc-sandbox.ucdenver.edu/~brentp/exome-region/info.bed">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;position=chr12%3A12022530-12022539&amp;hg.customText=http://amc-sandbox.ucdenver.edu/~brentp/exome-region/info.bed</a></div></blockquote>
</div>
<div class="section" id="viewing-in-igv">
<h2>Viewing in IGV<a class="headerlink" href="#viewing-in-igv" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#viewing-in-igv" title="Slides">§</a></h2>
<img alt="../_images/igv_snapshot.png" src="../_images/igv_snapshot.png" />
</div>
<div class="section" id="viewing-with-samtools-tview">
<h2>Viewing with samtools tview<a class="headerlink" href="#viewing-with-samtools-tview" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#viewing-with-samtools-tview" title="Slides">§</a></h2>
<div class="highlight-bash"><div class="highlight"><pre><span class="c">#module load samtools/0.1.19</span>

samtools tview -p 12:12022535 <span class="se">\</span>
    ~brentp/exomes/results/ll-1.dedup.bam <span class="se">\</span>
    ~brentp/data/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa
</pre></div>
</div>
</div>
<div class="section" id="extracting-parts-of-the-alignment">
<h2>Extracting parts of the alignment<a class="headerlink" href="#extracting-parts-of-the-alignment" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-26.html#extracting-parts-of-the-alignment" title="Slides">§</a></h2>
<p>When you only want to look at part of the alignment. No need to transfer
20GB of <cite>BAM</cite> files to your laptop</p>
<div class="highlight-bash"><div class="highlight"><pre>samtools view -h results/<span class="nv">$sample</span>.dedup.bam 12:12022035-12023035 <span class="se">\</span>
        &gt; ~/cpy/save.<span class="nv">$sample</span>.bam
</pre></div>
</div>
<p>will save on the portion that we are interested in.</p>
</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
        <br/>
        
<div id="sourcelink">
  <a href="../_sources/Classes/class-26.txt"
     rel="nofollow">Source</a>
</div>
      
    </p>
    <p>
        &copy; Copyright 2014, Jay Hesselberth.<br/>
      Last updated on Aug 28, 2014.<br/>
    </p>
  </div>
</footer>
  </body>
</html>