<!--
Google IO 2012/2013 HTML5 Slide Template

Authors: Eric Bidelman <ebidel@gmail.com>
         Luke Mahé <lukem@google.com>

URL: https://code.google.com/p/io-2012-slides
--><!DOCTYPE html>


<html>
<head>
  <title>Class 14 : ChIP-seq &mdash; Genome Informatics Workshop 0.1 documentation</title>
  <meta charset="utf-8">
  
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">-->
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0">-->
  <!--This one seems to work all the time, but really small on ipad-->
  <!--<meta name="viewport" content="initial-scale=0.4">-->
  <meta name="apple-mobile-web-app-capable" content="yes">
  
  <link rel="stylesheet" media="all"
        href="../_static/theme/css/default.css">
  <link rel="stylesheet" media="only screen and (max-device-width: 480px)"
        href="../_static/theme/css/phone.css">

    

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->
  
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>

    <script data-main="../_static/js/slides"
            src="../_static/js/require-1.0.8.min.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="top" title="Genome Informatics Workshop 0.1 documentation" href="../index.html" />
    <link rel="up" title="Class list" href="../Classes.html" />
    <link rel="next" title="Class 15 : ChIP-seq / DNase I" href="class-15.html" />
    <link rel="prev" title="Class 13 : BEDtools" href="class-13.html" /> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>


</head>
<body style="opacity: 0">

<slides class="layout-widescreen">

    <slide class="title-slide segue nobackground">
    <aside class="gdbar"><img src="images/google_developers_icon_128.png"></aside>
    <!-- The content of this hgroup is replaced programmatically through the slide_config.json. -->
    <hgroup class="auto-fadein">
      <h1 data-config-title><!-- populated from slide_config.json --></h1>
      <h2 data-config-subtitle><!-- populated from slide_config.json --></h2>
      <p data-config-presenter><!-- populated from slide_config.json --></p>
    </hgroup>
  </slide>

  
    <slide class="level-1">
    <hgroup>
      <h1>Class 14 : ChIP-seq<a class="headerlink" href="../../Classes/class-14.html#class-14-chip-seq" title="View HTML">§</a></h1>
    </hgroup>
    <article class="">
      <table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Class date:</th><td class="field-body">2014 Feb 28 Friday</td>
</tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Goals<a class="headerlink" href="../../Classes/class-14.html#goals" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ol class="arabic simple">
<li>Learn the workflow for analyzing ChIP-seq data</li>
</ol>
</div></blockquote>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p><strong>LOGIN TO THE CLUSTER</strong></p>
<p class="last">You will need access to amc-tesla to do your homework. Confirm you can
log on before leaving class.</p>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Chromatin Immunoprecipitation Overview<a class="headerlink" href="../../Classes/class-14.html#chromatin-immunoprecipitation-overview" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Chromatin Immunoprecipitation is used to determine where a protein of
interest binds on a chromatin template <a class="reference internal" href="#park-chipseq" id="id1">[Park_Chipseq]</a>.</p>
<table class="docutils citation" frame="void" id="park-chipseq" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[Park_Chipseq]</a></td><td><a class="reference external" href="http://www.nature.com/nrg/journal/v10/n10/full/nrg2641.html">http://www.nature.com/nrg/journal/v10/n10/full/nrg2641.html</a></td></tr>
</tbody>
</table>
<img alt="../_images/chip-workflow.png" src="../_images/chip-workflow.png" />




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Chromatin Immunoprecipitation Overview<a class="headerlink" href="../../Classes/class-14.html#id8" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <img alt="../_images/chip-data.png" src="../_images/chip-data.png" />




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>ChIP-seq analysis workflow<a class="headerlink" href="../../Classes/class-14.html#chip-seq-analysis-workflow" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>A general workflow for visualizing ChIP-seq data (and many other types of
data) is:</p>
<table border="1" class="docutils">
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Operation</th>
<th class="head">File formats</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Align reads to reference genome</td>
<td><tt class="docutils literal"><span class="pre">FASTQ</span> <span class="pre">~~&gt;</span> <span class="pre">BAM</span></tt></td>
</tr>
<tr class="row-odd"><td>Generate coverage plots</td>
<td><tt class="docutils literal"><span class="pre">BAM</span> <span class="pre">~~&gt;</span> <span class="pre">bedGraph</span></tt></td>
</tr>
<tr class="row-even"><td>Call peaks</td>
<td><tt class="docutils literal"><span class="pre">BAM</span> <span class="pre">~~&gt;</span> <span class="pre">BED</span></tt></td>
</tr>
<tr class="row-odd"><td>Make binary files for UCSC display</td>
<td><tt class="docutils literal"><span class="pre">bedGraph</span> <span class="pre">~~&gt;</span> <span class="pre">bigWig</span></tt>, <tt class="docutils literal"><span class="pre">BED</span> <span class="pre">~~&gt;</span> <span class="pre">bigBed</span></tt></td>
</tr>
<tr class="row-even"><td>Identify motifs</td>
<td><tt class="docutils literal"><span class="pre">BED</span> <span class="pre">~~&gt;</span> <span class="pre">FASTA</span> <span class="pre">~~&gt;</span> <span class="pre">TXT</span> <span class="pre">/</span> <span class="pre">HTML</span></tt></td>
</tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>ChIP-seq data<a class="headerlink" href="../../Classes/class-14.html#chip-seq-data" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Look at some human ChIP-seq data <a class="footnote-reference" href="#id3" id="id2">[1]</a>.</p>
<table class="docutils footnote" frame="void" id="id3" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[1]</a></td><td>Genome Browser Session
<a class="reference external" href="http://goo.gl/WfJxcM">http://goo.gl/WfJxcM</a></td></tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Short read alignment<a class="headerlink" href="../../Classes/class-14.html#short-read-alignment" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>There are several short read alignment packages available. We will use
bowtie2 <a class="footnote-reference" href="#id5" id="id4">[2]</a> because it is easy to use and relatively fast.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># minimal bowtie2 command</span>
<span class="nv">$ </span>bowtie2 -x &lt;index&gt; -U &lt;read1.fq.gz&gt; <span class="o">[</span>options<span class="o">]</span> &gt; output.sam
</pre></div>
</div>
<table class="docutils footnote" frame="void" id="id5" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id4">[2]</a></td><td>Bowtie2
<a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a></td></tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Short read alignment<a class="headerlink" href="../../Classes/class-14.html#id9" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Normally, you would also pipe the SAM-format alignment through the samtools
suite to discard unaligned reads, sort the alignment and store it in
binary format (bam).</p>
<p>SAM format: <a class="reference external" href="http://samtools.sourceforge.net/SAMv1.pdf">http://samtools.sourceforge.net/SAMv1.pdf</a></p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># sort sam output with samtools and write bam output</span>
<span class="nv">$ </span>bowtie2 -x &lt;index&gt; -U &lt;read1.fq.gz&gt; <span class="o">[</span>options<span class="o">]</span> <span class="se">\</span>
    | samtools view -ShuF4 - <span class="se">\</span>
    | samtools sort -o - aln.temp -m 8G <span class="se">\</span>
    &gt; aln.bam
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Generate and visualize coverage plots<a class="headerlink" href="../../Classes/class-14.html#generate-and-visualize-coverage-plots" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Once alignment is complete, you can create coverage plots from your aligned
data, so that you can visualize your data.</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>You will need a &quot;chromsizes&quot; flle for many BEDtools commands. This file
contains the sizes of each chromosome in an assembly. UCSC provides a
tool to retrieve this information:</p>
<div class="last highlight-bash"><div class="highlight"><pre><span class="c"># retrieve chrom sizes for the hg19 assembly and write them to a file</span>
<span class="c"># inspect this file so you know what it looks like</span>
<span class="nv">$ </span>fetchChromSizes hg19 &gt; hg19.chrom.sizes
</pre></div>
</div>
</div>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Coverage plots with BEDtools<a class="headerlink" href="../../Classes/class-14.html#coverage-plots-with-bedtools" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <p>To generate coverage plots, we will use BEDtools. Here, we'll
use the <a class="reference external" href="http://bedtools.readthedocs.org/en/latest/content/tools/genomecov.html#genomecov" title="(in bedtools v2.20.1)"><em class="xref std std-ref">genomecov</em></a> tool.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># -bg : write ouptput in bedGraph format</span>
<span class="nv">$ </span>bedtools genomecov -i &lt;aln.bam&gt; -g &lt;chrom.sizes&gt; -bg &gt; coverage.bg
</pre></div>
</div>
<p>This command writes a bedGraph format file called <tt class="docutils literal"><span class="pre">coverage.bg</span></tt>. Use
<tt class="docutils literal"><span class="pre">less</span></tt> to examine this file.</p>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Coverage plots with BEDtools (2)<a class="headerlink" href="../../Classes/class-14.html#id10" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <p>Words to live by: <strong>If you make a BED file, sort the BED file</strong></p>
<p>Many strange things can happen if you use unsorted BED files for
analysis. Once you create a BED file, sort it with one of these:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># using UNIX sort</span>
<span class="nv">$ </span>sort -k1,1 -k2,2 unsorted.bed &gt; sorted.bed

<span class="c"># UCSC tool. same filename twice, overwrites original file</span>
<span class="nv">$ </span>bedSort file.bed file.bed

<span class="c"># or you can use bedtools; writes additional file</span>
<span class="nv">$ </span>bedtools sort -i - &lt; unsorted.bed &gt; sorted.bed
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Coverage plots split by strand<a class="headerlink" href="../../Classes/class-14.html#coverage-plots-split-by-strand" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <p>For some experiments, you will analyze the data relative to each strand of
the reference genome. For example, RNA is transcribed in single-stranded
form and derives from one or the other strand.</p>
<p>During alignment, reads from an RNA-based experiment will map to either
the positive ('+' or <tt class="docutils literal"><span class="pre">pos</span></tt>) or negative ('-' or <tt class="docutils literal"><span class="pre">neg</span></tt>) strand. You can
generate signal plots for <tt class="docutils literal"><span class="pre">pos</span></tt> and <tt class="docutils literal"><span class="pre">neg</span></tt> strands separately with
<tt class="docutils literal"><span class="pre">bedtools</span></tt>:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ common_args</span><span class="o">=</span><span class="s2">&quot;-ibam &lt;aln.bam&gt; -g &lt;chrom.size&gt; -bg&quot;</span>
<span class="nv">$ </span>bedtools genomecov <span class="nv">$common_args</span> -strand + &gt; coverage.pos.bg
<span class="nv">$ </span>bedtools genomecov <span class="nv">$common_args</span> -strand - &gt; coverage.neg.bg
</pre></div>
</div>
<p>You would then create bigWigs for each of these display the stranded data
in the Genome Browser.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Peak calling<a class="headerlink" href="../../Classes/class-14.html#peak-calling" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>There are several available software packages for identifying regions
enriched in your IP experiment (i.e. peaks). We will use macs2 here.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># minimal macs2 command</span>
<span class="nv">$ </span>macs2 callpeak --treatment &lt;aln.bam&gt; --name &lt;exp.name&gt; <span class="o">[</span>options<span class="o">]</span>
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Identify sequence motifs in enriched regions<a class="headerlink" href="../../Classes/class-14.html#identify-sequence-motifs-in-enriched-regions" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>You can use meme <a class="footnote-reference" href="#id7" id="id6">[3]</a> to identify over-represented motifs in groups of
sequences (e.g. sequences covered by ChIP peaks).</p>
<p>Use the <a class="reference external" href="http://bedtools.readthedocs.org/en/latest/content/tools/getfasta.html#getfasta" title="(in bedtools v2.20.1)"><em class="xref std std-ref">bedtools getfasta</em></a> command to fetch
fasta sequences.</p>
<p>Note: meme looks at both strands of a DNA sequence by default.</p>
<table class="docutils footnote" frame="void" id="id7" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id6">[3]</a></td><td>MEME
<a class="reference external" href="http://meme.nbcr.net/meme/">http://meme.nbcr.net/meme/</a></td></tr>
</tbody>
</table>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>bedtools getfasta -fi &lt;ref.fa&gt; -bed &lt;peaks.bed&gt; -fo peaks.fa
<span class="nv">$ </span>meme -nmotifs 5 -minw 6 -maxw 20 -dna &lt;peaks.fa&gt;
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Putting it all together<a class="headerlink" href="../../Classes/class-14.html#putting-it-all-together" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Here is a script that combines the above in a single workflow:</p>
<div class="highlight-bash"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79</pre></div></td><td class="code"><div class="highlight"><pre><span class="c">#! /usr/bin/env bash</span>

<span class="c">#BSUB -J chipseq</span>
<span class="c">#BSUB -o %J.out</span>
<span class="c">#BSUB -e %J.err</span>

<span class="c">#</span>
<span class="c"># load on the head-node required modules before submitting this script</span>
<span class="c">#</span>
<span class="c"># $ module load modules modules-init modules-python</span>
<span class="c"># $ module load python</span>
<span class="c"># $ module load bowtie2</span>
<span class="c"># $ module load ucsc</span>
<span class="c"># $ module load bedtools</span>
<span class="c"># $ module load meme</span>
<span class="c">#</span>

<span class="nb">set</span> -e

<span class="nv">data</span><span class="o">=</span>/vol1/opt/data
<span class="nv">fasta</span><span class="o">=</span><span class="nv">$data</span>/hg19.fa
<span class="nv">bwtindex</span><span class="o">=</span><span class="nv">$data</span>/hg19
<span class="nv">chromsize</span><span class="o">=</span><span class="nv">$data</span>/hg19.chrom.sizes

<span class="c"># downloaded from ENCODE website - H3K4me3 ChIP in Hela cells</span>
<span class="nv">fastqfile</span><span class="o">=</span><span class="nv">$data</span>/wgEncodeBroadHistoneHelas3H3k4me3StdRawDataRep1.fastq.gz

<span class="nv">expname</span><span class="o">=</span><span class="s2">&quot;chipseq.hela.h3k4me3&quot;</span>
<span class="nv">bamfile</span><span class="o">=</span><span class="nv">$expname</span>.bam

<span class="c"># run the alignment</span>
bowtie2 -x <span class="nv">$bwtindex</span> -U <span class="nv">$fastqfile</span> <span class="se">\</span>
    | samtools view -ShuF4 - <span class="se">\</span>
    | samtools sort -o - aln.temp -m 8G <span class="se">\</span>
    &gt; <span class="nv">$bamfile</span>

<span class="c"># call peaks</span>
macs2 callpeak --treatment <span class="nv">$bamfile</span> --name <span class="nv">$expname</span>

<span class="c"># macs2 will generate a series of files, including one in</span>
<span class="c"># ``narrowPeak`` format, an extended BED fromat:</span>
<span class="nv">peakbed</span><span class="o">=</span>chipseq.hela.h3k4me3_peaks.narrowPeak
<span class="nv">peakbigbed</span><span class="o">=</span>chipseq.hela.h3k4me3_peaks.bb
bedToBigBed <span class="nv">$peakbed</span> <span class="nv">$chromsize</span> <span class="nv">$peakbigbed</span>

<span class="c"># make coverage plots in bedgraph and bigwig</span>
<span class="nv">bedgraph</span><span class="o">=</span><span class="s2">&quot;$expname.bg&quot;</span>
<span class="nv">bigwig</span><span class="o">=</span><span class="s2">&quot;$expname.bw&quot;</span>
bedtools genomecov -ibam <span class="nv">$bamfile</span> -g <span class="nv">$chromsize</span> -bg &gt; <span class="nv">$bedgraph</span>
bedGraphToBigWig <span class="nv">$bedgraph</span> <span class="nv">$chromsize</span> <span class="nv">$bigwig</span>

<span class="c"># find motifs on chr22</span>
<span class="nv">peakfasta</span><span class="o">=</span><span class="nv">$expname</span>.peaks.fa
<span class="nv">peak_chr22_fasta</span><span class="o">=</span><span class="nv">$expname</span>.peaks.chr22.fa
bedtools getfasta -fi <span class="nv">$fasta</span> -bed <span class="nv">$peakbed</span> -fo <span class="nv">$peakfasta</span>
awk <span class="s1">&#39;$1 == &quot;chr22&quot;&#39;</span> &lt; <span class="nv">$peakfasta</span> &gt; <span class="nv">$peak_chr22_fasta</span>

<span class="c"># XXX this will take a while to run ...</span>
<span class="nv">meme_args</span><span class="o">=</span><span class="s2">&quot;-nmotifs 5 -minw 6 -maxw 20 -maxsize 10000000 -dna&quot;</span>
meme <span class="nv">$meme_args</span> <span class="nv">$peak_chr22_fasta</span>

<span class="c"># cp the bigWig / bigBed to public_html directory</span>
<span class="c">#</span>
<span class="c"># bigWig trackline (needs to be all on one line)</span>
<span class="c">#</span>
<span class="c"># track type=bigWig</span>
<span class="c"># name=&#39;coverage&#39;</span>
<span class="c"># bigDataUrl=&quot;http://amc-sandbox.ucdenver.edu/~username/file.bw</span>
<span class="c"># visibility=full</span>
<span class="c"># maxHeightPixels=50:35:35</span>
<span class="c">#</span>
<span class="c"># bigBed trackline</span>
<span class="c">#</span>
<span class="c"># track type=bigBed</span>
<span class="c"># name=&#39;peaks&#39;</span>
<span class="c"># bigDataUrl=&quot;http://amc-sandbox.ucdenver.edu/~username/file.bb</span>
<span class="c"># visibility=dense</span>
<span class="c"># maxHeightPixels=50:35:35</span>
<span class="c">#</span>
</pre></div>
</td></tr></table></div>




    </article>
  </slide>


    <slide class="thank-you-slide segue nobackground">
    <aside class="gdbar right"><img src="images/google_developers_icon_128.png"></aside>
    <article class="flexbox vleft auto-fadein">
      <h2>&lt;Thank You!&gt;</h2>
    </article>
    <p class="auto-fadein" data-config-contact>
      <!-- populated from slide_config.json -->
    </p>
  </slide>

  <slide class="backdrop"></slide>

</slides>

<!--[if IE]>
  <script src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js"></script>
  <script>CFInstall.check({mode: 'overlay'});</script>
<![endif]-->
</body>
</html>