<!--
Google IO 2012/2013 HTML5 Slide Template

Authors: Eric Bidelman <ebidel@gmail.com>
         Luke Mahé <lukem@google.com>

URL: https://code.google.com/p/io-2012-slides
--><!DOCTYPE html>


<html>
<head>
  <title>Class 6 : Working in a cluster environment (part 2) &mdash; Genome Informatics Workshop 0.2 documentation</title>
  <meta charset="utf-8">
  
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">-->
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0">-->
  <!--This one seems to work all the time, but really small on ipad-->
  <!--<meta name="viewport" content="initial-scale=0.4">-->
  <meta name="apple-mobile-web-app-capable" content="yes">
  
  <link rel="stylesheet" media="all"
        href="../_static/theme/css/default.css">
  <link rel="stylesheet" media="all"
        href="../_static/theme/css/hieroglyph.css">
  <link rel="stylesheet" media="only screen and (max-device-width: 480px)"
        href="../_static/theme/css/phone.css">

    

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->
  
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>

    <script data-main="../_static/js/slides"
            src="../_static/js/require-1.0.8.min.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="top" title="Genome Informatics Workshop 0.2 documentation" href="../index.html" />
    <link rel="up" title="Class list" href="../Classes.html" />
    <link rel="next" title="Class 7 : Python : Basics" href="class-7.html" />
    <link rel="prev" title="Class 5 : Working in a cluster environment" href="class-5.html" /> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>


</head>
<body style="opacity: 0">

<slides class="layout-widescreen">

    <slide class="title-slide segue nobackground">
    <aside class="gdbar"><img src="images/google_developers_icon_128.png"></aside>
    <!-- The content of this hgroup is replaced programmatically through the slide_config.json. -->
    <hgroup class="auto-fadein">
      <h1 data-config-title><!-- populated from slide_config.json --></h1>
      <h2 data-config-subtitle><!-- populated from slide_config.json --></h2>
      <p data-config-presenter><!-- populated from slide_config.json --></p>
    </hgroup>
  </slide>

  
    <slide class="level-1">
    <hgroup>
      <h1>Class 6 : Working in a cluster environment (part 2)<a class="headerlink" href="../../Classes/class-6.html#class-6-working-in-a-cluster-environment-part-2" title="View HTML">§</a></h1>
    </hgroup>
    <article class="">
      <table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Class date:</th><td class="field-body">Friday 7 February 2014</td>
</tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Goals<a class="headerlink" href="../../Classes/class-6.html#goals" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <ol class="arabic simple">
<li>Leverage cluster resources to solve bioinformatic problems</li>
<li>Exercises! (<tt class="docutils literal"><span class="pre">grep</span></tt> review)</li>
<li>Practice typing for a bit.</li>
<li>Learn to switch between terminal windows with &lt;Alt&gt;-&lt;Tab&gt;</li>
</ol>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Note<a class="headerlink" href="../../Classes/class-6.html#note" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>We would like to quickly move toward having your homework be executable.
In other words, we should be able to:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>bash run.homework-1.sh
</pre></div>
</div>
<p>and have it execute all of the questions, writing results into files into
the right places as necessary. Getting into this habit lets you save
everything you did, and also re-run things when you need to make changes.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Overview<a class="headerlink" href="../../Classes/class-6.html#overview" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>We will go over a common scenario in bioinformatics where you can
leverage cluster resources to process many files simultaneously with
identical workflows.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Exercise<a class="headerlink" href="../../Classes/class-6.html#exercise" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Sequencing experiments typically generate raw sequencing data from many
different experimental conditions (i.e. treatments, controls, cell types,
time points, etc.) However, the initial steps of processing sequencing
data use common steps to process the data into an interpretable form. A
typical workflow for assessing sequencing data might begin with:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Align reads to a reference genome</li>
<li>Calculate coverage from the alignment</li>
</ol>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Exercise (2)<a class="headerlink" href="../../Classes/class-6.html#id1" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Login to amc-tesla.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># the -X flag starts an X11 connection</span>
<span class="nv">$ </span>ssh -X username@amc-tesla.ucdenver.pvt
</pre></div>
</div>
<p>We have 4 different FASTQ files from a sequencing experiment. They
are located in:</p>
<div class="highlight-python"><div class="highlight"><pre>/vol1/opt/data
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Exercise (3)<a class="headerlink" href="../../Classes/class-6.html#id2" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Type this out. Do not copy / paste.</p>
</div>
<div class="highlight-bash"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33</pre></div></td><td class="code"><div class="highlight"><pre><span class="c">#! /usr/bin/env bash</span>

<span class="c">#BSUB -J workflow[1-4]                                                                                                              </span>
<span class="c">#BSUB -e %J.%I.err                                                                                                                  </span>
<span class="c">#BSUB -o %J.%I.out                                                                                                                  </span>
<span class="c">#BSUB -R &quot;select[mem&gt;4] rusage[mem=4]&quot;</span>

<span class="c"># this section defines the sample names and grabs the appropriate                                                                   </span>
<span class="c"># sample name for the current process</span>
<span class="nv">SAMPLENAMES</span><span class="o">=(</span>SP1 SP2 SP3 SP4<span class="o">)</span>                                                                                                       

<span class="c"># $LSB_JOBINDEX is set at runtime for the current process; in this                                                                  </span>
<span class="c"># case you asked for 4 jobs, so it be a value between 1 and 4                                                                       </span>
<span class="nv">sample</span><span class="o">=</span><span class="k">${</span><span class="nv">SAMPLENAMES</span><span class="p">[</span><span class="k">$((</span><span class="nv">$LSB_JOBINDEX</span> <span class="o">-</span> <span class="m">1</span><span class="k">))</span><span class="p">]</span><span class="k">}</span>                                                                                       

<span class="c"># set up file names                                                                                                                 </span>
<span class="nv">data</span><span class="o">=</span>/vol1/opt/data
<span class="nv">bwtindex</span><span class="o">=</span>/vol1/opt/data/hg19
<span class="nv">fastq</span><span class="o">=</span><span class="nv">$data</span>/<span class="nv">$sample</span>.fq.gz
<span class="nv">align</span><span class="o">=</span><span class="nv">$sample</span>.sam
<span class="nv">coverage</span><span class="o">=</span><span class="nv">$sample</span>.coverage.gz

<span class="c"># run the workflow </span>
bowtie2 -x <span class="nv">$bwtindex</span> -U <span class="nv">$fastq</span> &gt; <span class="nv">$align</span>

<span class="c"># now generate counts of each position that was aligned to; fields 3                                                                </span>
<span class="c"># and 4 of the .sam file are chrom and pos.</span>
grep -v <span class="s1">&#39;^@&#39;</span> <span class="nv">$align</span> <span class="se">\</span>
    | cut -f3,4 <span class="se">\</span>
    | sort <span class="se">\</span>
    | uniq -c <span class="se">\</span>
    | awk <span class="s1">&#39;BEGIN {OFS=&quot;\t&quot;} {print $2,$3,$1}&#39;</span> <span class="se">\</span>
    | gzip -c &gt; <span class="nv">$coverage</span>
</pre></div>
</td></tr></table></div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Exercise (4)<a class="headerlink" href="../../Classes/class-6.html#id3" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Run the above script and check its status immediately:</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>bsub &lt; run.sh
<span class="nv">$ </span>bjobs
</pre></div>
</div>
<p>You should see 4 running jobs, each with its own index i.e. workflow[1],
workflow[2] etc.</p>
<p>These will run for a bit. After they are done, you will see several new
files, corresponding to the output of the same analysis applied to all the
different samples.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Questions<a class="headerlink" href="../../Classes/class-6.html#questions" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ol class="arabic">
<li><p class="first">Check the <tt class="docutils literal"><span class="pre">.err</span></tt> files from the run. What information do they
contain? What does this tell you about your starting sequences?</p>
</li>
<li><p class="first">Find out how you would modify the <tt class="docutils literal"><span class="pre">bowtie2</span></tt> command to write out the
unaligned reads into a new file. Re-run the analysis to report those
reads.</p>
</li>
<li><p class="first">Modify the <tt class="docutils literal"><span class="pre">awk</span></tt> command in the script to print out a valid BED4
format:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">chrom</span> <span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span> <span class="n">start</span> <span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span> <span class="n">end</span> <span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span> <span class="n">count</span>
</pre></div>
</div>
</li>
<li><p class="first">Find out how many unique UMI sequences are associted with each
chromosomal coordinate (note: not as easy).</p>
</li>
</ol>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>More exercises<a class="headerlink" href="../../Classes/class-6.html#more-exercises" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ol class="arabic simple">
<li>use <tt class="docutils literal"><span class="pre">grep</span></tt> to identify lines in lamina.bed where the second field
(start) begins with <tt class="docutils literal"><span class="pre">100</span></tt>.</li>
<li>use <tt class="docutils literal"><span class="pre">grep</span></tt> to identify lines in lamina.bed where the third field
(end) ends with 99 .</li>
<li>use <tt class="docutils literal"><span class="pre">grep</span></tt> with its <tt class="docutils literal"><span class="pre">-w</span></tt> flag to count the number of 'chr1'
records in lamina.bed.</li>
<li>use <tt class="docutils literal"><span class="pre">grep</span></tt> to count how many fastq records are in the
/vol1/opt/data/t_R1.fastq.gz file (fastq records begin with an
'&#64;' symbol)</li>
<li>login to amc-tesla. use <tt class="docutils literal"><span class="pre">grep</span></tt> to count the number of fastq records
in /vol1/opt/data/SP1.fq.gz</li>
</ol>
</div></blockquote>




    </article>
  </slide>


    <slide class="thank-you-slide segue nobackground">
    <aside class="gdbar right"><img src="images/google_developers_icon_128.png"></aside>
    <article class="flexbox vleft auto-fadein">
      <h2>&lt;Thank You!&gt;</h2>
    </article>
    <p class="auto-fadein" data-config-contact>
      <!-- populated from slide_config.json -->
    </p>
  </slide>

  <slide class="backdrop"></slide>

</slides>

<!--[if IE]>
  <script src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js"></script>
  <script>CFInstall.check({mode: 'overlay'});</script>
<![endif]-->
</body>
</html>