<!--
Google IO 2012/2013 HTML5 Slide Template

Authors: Eric Bidelman <ebidel@gmail.com>
         Luke Mahé <lukem@google.com>

URL: https://code.google.com/p/io-2012-slides
--><!DOCTYPE html>


<html>
<head>
  <title>Class 25 : Exome-Seq - Alignment &mdash; Genome Informatics Workshop 0.2 documentation</title>
  <meta charset="utf-8">
  
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">-->
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0">-->
  <!--This one seems to work all the time, but really small on ipad-->
  <!--<meta name="viewport" content="initial-scale=0.4">-->
  <meta name="apple-mobile-web-app-capable" content="yes">
  
  <link rel="stylesheet" media="all"
        href="../_static/theme/css/default.css">
  <link rel="stylesheet" media="only screen and (max-device-width: 480px)"
        href="../_static/theme/css/phone.css">

    

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->
  
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>

    <script data-main="../_static/js/slides"
            src="../_static/js/require-1.0.8.min.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="top" title="Genome Informatics Workshop 0.2 documentation" href="../index.html" />
    <link rel="up" title="Class list" href="../Classes.html" />
    <link rel="next" title="Class 26 : Exome-Seq - Variants" href="class-26.html" />
    <link rel="prev" title="Class 24 : RNA-Seq II" href="class-24.html" /> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>


</head>
<body style="opacity: 0">

<slides class="layout-widescreen">

    <slide class="title-slide segue nobackground">
    <aside class="gdbar"><img src="images/google_developers_icon_128.png"></aside>
    <!-- The content of this hgroup is replaced programmatically through the slide_config.json. -->
    <hgroup class="auto-fadein">
      <h1 data-config-title><!-- populated from slide_config.json --></h1>
      <h2 data-config-subtitle><!-- populated from slide_config.json --></h2>
      <p data-config-presenter><!-- populated from slide_config.json --></p>
    </hgroup>
  </slide>

  
    <slide class="level-1">
    <hgroup>
      <h1>Class 25 : Exome-Seq - Alignment<a class="headerlink" href="../../Classes/class-25.html#class-25-exome-seq-alignment" title="View HTML">§</a></h1>
    </hgroup>
    <article class="">
      



    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Goals<a class="headerlink" href="../../Classes/class-25.html#goals" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ol class="arabic simple">
<li>Not kill Tesla</li>
<li>Exome Sequencing</li>
<li>Alignment</li>
<li>QC Alignments</li>
</ol>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Exome Sequencing<a class="headerlink" href="../../Classes/class-25.html#exome-sequencing" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ul class="simple">
<li>captures coding regions of genome</li>
<li>deeper coverage and more samples at same cost</li>
<li>goal is to call variants (SNPs/deletions/insertions)</li>
<li>higher concentration of interesting variants
relative to non-coding genome</li>
</ul>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Project Design<a class="headerlink" href="../../Classes/class-25.html#project-design" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <img alt="../_images/phenotype.png" src="../_images/phenotype.png" />




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Project Design II<a class="headerlink" href="../../Classes/class-25.html#project-design-ii" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>8 sequenced samples from single family</p>
<blockquote>
<div><ul class="simple">
<li>paired-end 100 base reads</li>
<li>5 affected</li>
<li>3 unaffected</li>
</ul>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Analysis<a class="headerlink" href="../../Classes/class-25.html#analysis" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ul class="simple">
<li>How likely are we to find something?</li>
<li>Naive probability that 5 individuals have a variant and 3 do not (without
considering pedigree)?</li>
<li>What if we find 1 million SNPs total?</li>
<li>How can we filter?</li>
</ul>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Alignment<a class="headerlink" href="../../Classes/class-25.html#alignment" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ul class="simple">
<li>bwa mem to align the reads.</li>
<li>bwa mem very good aligner for long reads</li>
</ul>
<p><a class="reference external" href="http://arxiv.org/pdf/1303.3997v2.pdf">http://arxiv.org/pdf/1303.3997v2.pdf</a></p>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>BWA mem<a class="headerlink" href="../../Classes/class-25.html#bwa-mem" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <img alt="../_images/bwa-mem-roc.png" src="../_images/bwa-mem-roc.png" />




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Mark Duplicates<a class="headerlink" href="../../Classes/class-25.html#mark-duplicates" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Marking/removing duplicates is to remove the influence of PCR duplication on
variant calling.</p>
<blockquote>
<div><ul class="simple">
<li>Marking/Removing duplicates makes sense unless you have extremely deep coverage,
especially on paired-end reads.</li>
<li>Uses position of both reads of a pair to determine duplication--if many pairs
have the same start and sequence, all but 1 pair is marked as a duplicate.</li>
<li>Recommended by the variant caller that we will use.</li>
<li>To mark or not to mark? <a class="reference external" href="http://seqanswers.com/forums/showthread.php?t=6854">http://seqanswers.com/forums/showthread.php?t=6854</a></li>
</ul>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Read Groups<a class="headerlink" href="../../Classes/class-25.html#read-groups" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Your alignments should have read-groups.</p>
<p>These tell downstream tools what library and sample your data came from</p>
<p>Used for calling duplicates (repeats must be from same library)</p>
<p>With read-groups, can have multiple samples in a single BAM.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Variant Calling<a class="headerlink" href="../../Classes/class-25.html#variant-calling" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>At each site in the genome, given:</p>
<blockquote>
<div><ul class="simple">
<li>base calls</li>
<li>base qualities</li>
<li>mapping qualities</li>
<li>surrounding sequence</li>
<li>position in read</li>
<li>strand bias</li>
<li>all of the above for other samples</li>
</ul>
</div></blockquote>
<p>What is the genotype of each sample and the confidence in that genotype?</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Variant Calling Software<a class="headerlink" href="../../Classes/class-25.html#variant-calling-software" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>The most commonly used software for variant calling is GATK:
Genome Analysis Toolkit</p>
<p>It has a number of steps including base recalibration, indel realignment,
etc.</p>
<p>We will use <cite>freebayes</cite> which has been shown to perform as well/better
(and it is much easier to use): <a class="reference external" href="https://github.com/ekg/freebayes">https://github.com/ekg/freebayes</a></p>
<p><a class="reference external" href="http://bcbio.wordpress.com/2013/10/21/updated-comparison-of-variant-detection-methods-ensemble-freebayes-and-minimal-bam-preparation-pipelines/">http://bcbio.wordpress.com/2013/10/21/updated-comparison-of-variant-detection-methods-ensemble-freebayes-and-minimal-bam-preparation-pipelines/</a></p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Freebayes<a class="headerlink" href="../../Classes/class-25.html#freebayes" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p><a class="reference external" href="https://github.com/ekg/freebayes">https://github.com/ekg/freebayes</a></p>
<p><em>FreeBayes is haplotype-based, in the sense that it calls variants based on the
literal sequences of reads aligned to a particular target, not their precise
alignment.</em></p>
<p><em>FreeBayes uses short-read alignments (BAM files) for any number of individuals
from a population and a reference genome to determine the most-likely combination
of genotypes for the population at each position in the reference. It reports
positions which it finds putatively polymorphic in variant call file (VCF) format.</em></p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>VCF Format<a class="headerlink" href="../../Classes/class-25.html#vcf-format" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>File format for SNP/indel calls:
<a class="reference external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41">http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41</a></p>
<div class="highlight-python"><div class="highlight"><pre>CHROM      1
POS        2345544
ID         rs1234..
REF        C
ALT        TA
QUAL       55.2
FILTER     PASS
INFO       NS=3;DP=14;AF=0.5;DB
FORMAT     GT:GQ:DP:HQ
sample_1   0|0:48:1:51,51
...
sample_n
</pre></div>
</div>
<p>&quot;:&quot; delimited chunks from samples match the format in <cite>FORMAT</cite></p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Pipeline<a class="headerlink" href="../../Classes/class-25.html#pipeline" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Walk through .sh scripts on <em>amc-tesla</em></p>




    </article>
  </slide>


    <slide class="thank-you-slide segue nobackground">
    <aside class="gdbar right"><img src="images/google_developers_icon_128.png"></aside>
    <article class="flexbox vleft auto-fadein">
      <h2>&lt;Thank You!&gt;</h2>
    </article>
    <p class="auto-fadein" data-config-contact>
      <!-- populated from slide_config.json -->
    </p>
  </slide>

  <slide class="backdrop"></slide>

</slides>

<!--[if IE]>
  <script src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js"></script>
  <script>CFInstall.check({mode: 'overlay'});</script>
<![endif]-->
</body>
</html>