<!--
Google IO 2012/2013 HTML5 Slide Template

Authors: Eric Bidelman <ebidel@gmail.com>
         Luke Mahé <lukem@google.com>

URL: https://code.google.com/p/io-2012-slides
--><!DOCTYPE html>


<html>
<head>
  <title>Class 23 : RNA-Seq I &mdash; Genome Informatics Workshop 0.1 documentation</title>
  <meta charset="utf-8">
  
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">-->
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0">-->
  <!--This one seems to work all the time, but really small on ipad-->
  <!--<meta name="viewport" content="initial-scale=0.4">-->
  <meta name="apple-mobile-web-app-capable" content="yes">
  
  <link rel="stylesheet" media="all"
        href="../_static/theme/css/default.css">
  <link rel="stylesheet" media="only screen and (max-device-width: 480px)"
        href="../_static/theme/css/phone.css">

    

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->
  
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>

    <script data-main="../_static/js/slides"
            src="../_static/js/require-1.0.8.min.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="top" title="Genome Informatics Workshop 0.1 documentation" href="../index.html" />
    <link rel="up" title="Class list" href="../Classes.html" />
    <link rel="next" title="Class 24 : RNA-Seq II" href="class-24.html" />
    <link rel="prev" title="Class 22 : R : Data Manipulation &amp; Plotting" href="class-22.html" /> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>


</head>
<body style="opacity: 0">

<slides class="layout-widescreen">

    <slide class="title-slide segue nobackground">
    <aside class="gdbar"><img src="images/google_developers_icon_128.png"></aside>
    <!-- The content of this hgroup is replaced programmatically through the slide_config.json. -->
    <hgroup class="auto-fadein">
      <h1 data-config-title><!-- populated from slide_config.json --></h1>
      <h2 data-config-subtitle><!-- populated from slide_config.json --></h2>
      <p data-config-presenter><!-- populated from slide_config.json --></p>
    </hgroup>
  </slide>

  
    <slide class="level-1">
    <hgroup>
      <h1>Class 23 : RNA-Seq I<a class="headerlink" href="../../Classes/class-23.html#class-23-rna-seq-i" title="View HTML">§</a></h1>
    </hgroup>
    <article class="">
      



    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Goals<a class="headerlink" href="../../Classes/class-23.html#goals" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ol class="arabic simple">
<li>Why RNA-Seq</li>
<li>QC.</li>
<li>Align Reads</li>
</ol>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>RNA-Seq<a class="headerlink" href="../../Classes/class-23.html#rna-seq" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ul class="simple">
<li>We use RNA-Seq primarily to measure gene expression</li>
<li>We can also find transcript use (transcript-switching/alternative-splicing)</li>
<li>Fusion Transcripts</li>
<li>Possible to call variants (SNPs) on RNA-Seq data</li>
</ul>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Library Prep<a class="headerlink" href="../../Classes/class-23.html#library-prep" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Libraries were prepared by a standard RNA-seq protocol including:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Oligo-dT purification of polyadenylated mRNA</li>
<li>Reverse transcription with random nonamers</li>
<li>Second strand sythesis to create double stranded DNA (dsDNA)</li>
<li>Shearing of dsDNA to 300-600 bp</li>
<li>End polishing, adaptor ligation, PCR with indexed primers</li>
<li>Quantify libraries, Mix, sequence.</li>
</ol>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Example Data<a class="headerlink" href="../../Classes/class-23.html#example-data" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>We will look at ~10 million 75 base paired-end reads from 4 RNA-seq
libraries prepared from these <cite>C. elegans</cite> strains:</p>
<table border="1" class="docutils">
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Strain</th>
<th class="head">Treatment</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>RtcB wild-type</td>
<td>No treatment</td>
</tr>
<tr class="row-odd"><td>RtcB wild-type</td>
<td>Tunicamycin (induces UPR)</td>
</tr>
<tr class="row-even"><td>RtcB mutant</td>
<td>No treatment</td>
</tr>
<tr class="row-odd"><td>RtcB mutant</td>
<td>Tunicamycin (induces UPR)</td>
</tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Example Data (2)<a class="headerlink" href="../../Classes/class-23.html#id4" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>RtcB is an RNA ligase. RtcB wild-type and mutant strains were treated with
tunicamycin, which inhibits protein glycosylation in the endoplasmic
reticulum and triggers the &quot;unfolded protein reponse&quot;:</p>
<div class="highlight-python"><div class="highlight"><pre>ER stress ~~&gt;
    Ire1 multimerization ~~&gt;
        Ire1 ribonuclease domain cuts
        xbp-1 transcript to release intron ~~&gt;
            RtcB ligates xbp-1 exons together ~~&gt;
                Xbp-1 protein is made, which activates UPR-related genes
                in the nucleus.
</pre></div>
</div>
<p>The expectation is that RtcB mutants will be unable to ligate specific RNA
substrates together, thus the mRNA library in the mutant might contain
altered levels of RNA fragments relative to the wild-type. In particular,
xbp-1 <a class="footnote-reference" href="#id2" id="id1">[1]</a> fragments should accumulate in the mutant.</p>
<table class="docutils footnote" frame="void" id="id2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[1]</a></td><td><a class="reference external" href="http://www.wormbase.org/species/c_elegans/gene/WBGene00006959?query=xbp-1#0-9e-3">http://www.wormbase.org/species/c_elegans/gene/WBGene00006959?query=xbp-1#0-9e-3</a></td></tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Experimental Design<a class="headerlink" href="../../Classes/class-23.html#experimental-design" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <div class="admonition important">
<p class="first admonition-title">Important</p>
<p>Please do not do RNA-Seq on fewer than 6 samples (3 vs 3).</p>
<p class="last">Even in a pilot study.</p>
</div>
<p>See: <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/30/3/301">http://bioinformatics.oxfordjournals.org/content/30/3/301</a></p>
<p>Titled: <strong>RNA-seq differential expression studies: more sequence or more
replication?</strong></p>
<p>Answer: <strong>more replication</strong></p>
<p>Your bioinformatician will thank you.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Analysis Plan<a class="headerlink" href="../../Classes/class-23.html#analysis-plan" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ol class="arabic simple">
<li>QC reads with Fastqc</li>
<li>Trim reads by quality with sickle</li>
<li>Re-check QC on trimmed reads</li>
<li>Align reads with Tophat<ol class="arabic">
<li>give a GTF to help it to find known transcripts</li>
<li>also allow novel junctions</li>
<li>fusion transcripts</li>
</ol>
</li>
<li>Count reads by gene with featureCounts (from subread)</li>
<li>Differential Expression statistics with DESeq2</li>
<li>Differential Transcript Use and fusion transcripts (if time)</li>
</ol>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>QC Reads<a class="headerlink" href="../../Classes/class-23.html#qc-reads" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>We will use fastqc to check the quality of the reads and look for adapter
contamination</p>
<blockquote>
<div>$ fastqc -o ~/public_html/fqc-$sample $data/${sample}_R2.fastq.gz</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Quality Trim Reads<a class="headerlink" href="../../Classes/class-23.html#quality-trim-reads" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>We will trim low-quality bases from reads using sickle:
<a class="reference external" href="https://github.com/najoshi/sickle">https://github.com/najoshi/sickle</a></p>
<p>Lenient trimming has been shown to improve mapping rates in
RNA-Seq. See, e.g. <a class="reference external" href="http://genomebio.org/is-trimming-is-beneficial-in-rna-seq/">http://genomebio.org/is-trimming-is-beneficial-in-rna-seq/</a></p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>QC Quality Trimmed Reads<a class="headerlink" href="../../Classes/class-23.html#qc-quality-trimmed-reads" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Compare before/after. Makes most difference in bad datasets.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Align with Tophat2<a class="headerlink" href="../../Classes/class-23.html#align-with-tophat2" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Many RNA-Seq (splice-aware) aligners:</p>
<blockquote>
<div><ul class="simple">
<li>Tophat/Tophat2</li>
<li>GSNAP</li>
<li>STAR</li>
<li>RUM</li>
<li>MapSplice</li>
<li>etc.</li>
</ul>
</div></blockquote>
<p>We will use tophat2.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>IGenomes<a class="headerlink" href="../../Classes/class-23.html#igenomes" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>If you work with a model organism, you can get bowtie (tophat2) genome
indexes and feature annotation in normalized format from Illumina.</p>
<p><a class="reference external" href="http://tophat.cbcb.umd.edu/igenomes.shtml">http://tophat.cbcb.umd.edu/igenomes.shtml</a></p>
<p>We will use the <em>C. elegans</em> data downloaded to <cite>amc-tesla:~brentp/data/ce/</cite></p>
<blockquote>
<div><ul class="simple">
<li>Using this will save you a lot of trouble</li>
<li>Tophat2 uses known transcripts and attempts to align to those as well as to
novel transcripts</li>
</ul>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Spliced Alignment<a class="headerlink" href="../../Classes/class-23.html#spliced-alignment" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>From tophat paper:</p>
<img alt="http://bioinformatics.oxfordjournals.org/content/25/9/1105/F1.large.jpg" src="http://bioinformatics.oxfordjournals.org/content/25/9/1105/F1.large.jpg" />




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Spliced Alignment<a class="headerlink" href="../../Classes/class-23.html#id3" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>From tophat2 paper:</p>
<img alt="http://genomebiology.com/content/figures/gb-2013-14-4-r36-1-l.jpg" src="http://genomebiology.com/content/figures/gb-2013-14-4-r36-1-l.jpg" />




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Tophat2 Invocation<a class="headerlink" href="../../Classes/class-23.html#tophat2-invocation" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <div class="highlight-bash"><div class="highlight"><pre>tophat2 -o <span class="nv">$out</span>/results/<span class="nv">$sample</span> <span class="nv">$reference</span> <span class="nv">$fq_1</span> <span class="nv">$fq2</span> <span class="se">\</span>
    --fusion-search -p 6 --transcriptome_index <span class="nv">$TINDEX</span> <span class="se">\</span>
    --GTF <span class="nv">$ANNOTATION_GTF</span>
</pre></div>
</div>
<p>Output will be in accepted_hits.bam</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Just Do It<a class="headerlink" href="../../Classes/class-23.html#just-do-it" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>On amc-tesla</p>
<div class="highlight-bash"><div class="highlight"><pre>mkdir -p ~/class-23/src/
cp ~brentp/class-23/run.sh ~/class-23/
cp ~brentp/class-23/src/clean-counts.py ~/class-23/src/
<span class="nb">cd</span> ~/class-23/src/
module load sickle
module load subread
module load bowtie2
</pre></div>
</div>




    </article>
  </slide>


    <slide class="thank-you-slide segue nobackground">
    <aside class="gdbar right"><img src="images/google_developers_icon_128.png"></aside>
    <article class="flexbox vleft auto-fadein">
      <h2>&lt;Thank You!&gt;</h2>
    </article>
    <p class="auto-fadein" data-config-contact>
      <!-- populated from slide_config.json -->
    </p>
  </slide>

  <slide class="backdrop"></slide>

</slides>

<!--[if IE]>
  <script src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js"></script>
  <script>CFInstall.check({mode: 'overlay'});</script>
<![endif]-->
</body>
</html>