<!--
Google IO 2012/2013 HTML5 Slide Template

Authors: Eric Bidelman <ebidel@gmail.com>
         Luke Mahé <lukem@google.com>

URL: https://code.google.com/p/io-2012-slides
--><!DOCTYPE html>


<html>
<head>
  <title>Class 15 : ChIP-seq / DNase I &mdash; Genome Informatics Workshop 0.2 documentation</title>
  <meta charset="utf-8">
  
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">-->
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0">-->
  <!--This one seems to work all the time, but really small on ipad-->
  <!--<meta name="viewport" content="initial-scale=0.4">-->
  <meta name="apple-mobile-web-app-capable" content="yes">
  
  <link rel="stylesheet" media="all"
        href="../_static/theme/css/default.css">
  <link rel="stylesheet" media="all"
        href="../_static/theme/css/hieroglyph.css">
  <link rel="stylesheet" media="only screen and (max-device-width: 480px)"
        href="../_static/theme/css/phone.css">

    

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->
  
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>

    <script data-main="../_static/js/slides"
            src="../_static/js/require-1.0.8.min.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="top" title="Genome Informatics Workshop 0.2 documentation" href="../index.html" />
    <link rel="up" title="Class list" href="../Classes.html" />
    <link rel="next" title="Class 16: Genomic Analysis Vignettes" href="class-16.html" />
    <link rel="prev" title="Class 14 : ChIP-seq" href="class-14.html" /> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>


</head>
<body style="opacity: 0">

<slides class="layout-widescreen">

    <slide class="title-slide segue nobackground">
    <aside class="gdbar"><img src="images/google_developers_icon_128.png"></aside>
    <!-- The content of this hgroup is replaced programmatically through the slide_config.json. -->
    <hgroup class="auto-fadein">
      <h1 data-config-title><!-- populated from slide_config.json --></h1>
      <h2 data-config-subtitle><!-- populated from slide_config.json --></h2>
      <p data-config-presenter><!-- populated from slide_config.json --></p>
    </hgroup>
  </slide>

  
    <slide class="level-1">
    <hgroup>
      <h1>Class 15 : ChIP-seq / DNase I<a class="headerlink" href="../../Classes/class-15.html#class-15-chip-seq-dnase-i" title="View HTML">§</a></h1>
    </hgroup>
    <article class="">
      <table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Class date:</th><td class="field-body">2014 Mar 3 Monday</td>
</tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Final Project<a class="headerlink" href="../../Classes/class-15.html#final-project" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>We will have a final project that is worth 15% of your grade, due on April
2.</p>
<p>We want you to come up with a question you can address with the tools you
learned in the class. You should incorporate ENCODE data into your
analysis, and you can also integrate your own data.</p>
<p>We will go over some analysis vignettes on Wed and Fri to illustrate
example final projects.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Goals<a class="headerlink" href="../../Classes/class-15.html#goals" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <blockquote>
<div><ol class="arabic simple">
<li>Connect UCSC browser to bigWig / bigBed files</li>
<li>Compare ChIP and DNase I data</li>
</ol>
</div></blockquote>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Coverage plots for ChIP and DNase I<a class="headerlink" href="../../Classes/class-15.html#coverage-plots-for-chip-and-dnase-i" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>You will want to calculate coverage plots that are appropriate for the
type of experiment. For example, in a ChIP experiment, you want to examine
the entire region covered by sequences.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ common_args</span><span class="o">=</span><span class="s2">&quot;-ibam &lt;aln.bam&gt; -g &lt;chrom.size&gt; -bg&quot;</span>
<span class="nv">$ </span>bedtools genomecov <span class="nv">$common_args</span> &gt; coverage.bg
</pre></div>
</div>
<p>But in a DNase I mapping experiment, you only want to know the exact
position where DNase I cut the DNA, i.e. the 5' position. Use the <tt class="docutils literal"><span class="pre">-5</span></tt>
flag in bedtools to only count 5' positions.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ common_args</span><span class="o">=</span><span class="s2">&quot;-ibam &lt;aln.bam&gt; -g &lt;chrom.size&gt; -bg&quot;</span>
<span class="nv">$ </span>bedtools genomecov <span class="nv">$common_args</span> -5 &gt; coverage.5p.bg
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>ChIP-seq / DNase I data<a class="headerlink" href="../../Classes/class-15.html#chip-seq-dnase-i-data" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Compare ChIP-seq and DNase I data in browser <a class="footnote-reference" href="#id2" id="id1">[1]</a>.</p>
<table class="docutils footnote" frame="void" id="id2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[1]</a></td><td>Genome Browser Session <a class="reference external" href="http://goo.gl/jx6fCA">http://goo.gl/jx6fCA</a></td></tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Plot coverage with the Genome Browser<a class="headerlink" href="../../Classes/class-15.html#plot-coverage-with-the-genome-browser" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <p>Use the UCSC Genome Browser to plot your data. Files in bedGraph format
can be large, so UCSC created a facility for posting binary format data in
a web-accessible directory that the browser can read.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># NOTE: bedGraph and BED files must be sorted first</span>

<span class="c"># convert bedGraph to binary format (bigWig)</span>
<span class="nv">$ </span>bedGraphToBigWig &lt;coverage.bg&gt; &lt;chrom.sizes&gt; &lt;coverage.bw&gt;

<span class="c"># convert BED to binary format (bigBed)</span>
<span class="nv">$ </span>bedToBigBed &lt;peaks.bed&gt; &lt;chrom.sizes&gt; &lt;peaks.bb&gt;
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Posting your data<a class="headerlink" href="../../Classes/class-15.html#posting-your-data" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <p>We need to set up your accounts so that you will be able to see data
posted in your account. When this is setup, you should be able to put
files in:</p>
<div class="highlight-python"><div class="highlight"><pre>$HOME/public_html/file.html
</pre></div>
</div>
<p>and be able to see them like:</p>
<div class="highlight-python"><div class="highlight"><pre>http://amc-sandbox.ucdenver.edu/~username/file.html
</pre></div>
</div>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Writing tracklines<a class="headerlink" href="../../Classes/class-15.html#writing-tracklines" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <p>You can now write &quot;tracklines&quot; <a class="footnote-reference" href="#id4" id="id3">[2]</a> to tell where UCSC to find your data:</p>
<div class="highlight-python"><div class="highlight"><pre># URL = http://amc-sandbox.ucdenver.edu/~username/path-to-binaryfile
track type=bigWig bigDataUrl=&lt;URL&gt; name=&#39;coverage&#39; color=r,g,b
track type=bigBed bigDataUrl=&lt;URL&gt; name=&#39;peaks&#39; color=r,g,b
</pre></div>
</div>
<table class="docutils footnote" frame="void" id="id4" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id3">[2]</a></td><td>UCSC Track configuration
<a class="reference external" href="https://genome.ucsc.edu/goldenPath/help/customTrack.html#TRACK">https://genome.ucsc.edu/goldenPath/help/customTrack.html#TRACK</a></td></tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Writing tracklines<a class="headerlink" href="../../Classes/class-15.html#id5" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>Don't pick colors yourself, they will be ugly. <strong>Use Colorbrewer</strong>
<a class="reference external" href="http://colorbrewer2.org">http://colorbrewer2.org</a>.</p>
<p class="last">RGB colors in the <tt class="docutils literal"><span class="pre">Dark2</span></tt> and <tt class="docutils literal"><span class="pre">Set1</span></tt> qualitative palettes work
well for UCSC display.</p>
</div>
<p>There are a large number of additional options you can use in tracklines
to change their display.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Problem Set Questions<a class="headerlink" href="../../Classes/class-15.html#problem-set-questions" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Anybody have questions on the problem set?</p>




    </article>
  </slide>


    <slide class="thank-you-slide segue nobackground">
    <aside class="gdbar right"><img src="images/google_developers_icon_128.png"></aside>
    <article class="flexbox vleft auto-fadein">
      <h2>&lt;Thank You!&gt;</h2>
    </article>
    <p class="auto-fadein" data-config-contact>
      <!-- populated from slide_config.json -->
    </p>
  </slide>

  <slide class="backdrop"></slide>

</slides>

<!--[if IE]>
  <script src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js"></script>
  <script>CFInstall.check({mode: 'overlay'});</script>
<![endif]-->
</body>
</html>