<!--
Google IO 2012/2013 HTML5 Slide Template

Authors: Eric Bidelman <ebidel@gmail.com>
         Luke Mahé <lukem@google.com>

URL: https://code.google.com/p/io-2012-slides
--><!DOCTYPE html>


<html>
<head>
  <title>Problem Set 3 &mdash; Genome Informatics Workshop 0.2 documentation</title>
  <meta charset="utf-8">
  
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0">-->
  <!--<meta name="viewport" content="width=device-width, initial-scale=1.0">-->
  <!--This one seems to work all the time, but really small on ipad-->
  <!--<meta name="viewport" content="initial-scale=0.4">-->
  <meta name="apple-mobile-web-app-capable" content="yes">
  
  <link rel="stylesheet" media="all"
        href="../../_static/theme/css/default.css">
  <link rel="stylesheet" media="only screen and (max-device-width: 480px)"
        href="../../_static/theme/css/phone.css">

    

    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->
  
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '0.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>

    <script data-main="../../_static/js/slides"
            src="../../_static/js/require-1.0.8.min.js"></script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="top" title="Genome Informatics Workshop 0.2 documentation" href="../../index.html" />
    <link rel="up" title="Problem Sets" href="../../Problem_Sets.html" />
    <link rel="next" title="Problem Set 4" href="../problem_set_4/problem_set_4.html" />
    <link rel="prev" title="Problem Set 2" href="../problem_set_2/problem_set_2.html" /> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>


</head>
<body style="opacity: 0">

<slides class="layout-widescreen">

    <slide class="title-slide segue nobackground">
    <aside class="gdbar"><img src="images/google_developers_icon_128.png"></aside>
    <!-- The content of this hgroup is replaced programmatically through the slide_config.json. -->
    <hgroup class="auto-fadein">
      <h1 data-config-title><!-- populated from slide_config.json --></h1>
      <h2 data-config-subtitle><!-- populated from slide_config.json --></h2>
      <p data-config-presenter><!-- populated from slide_config.json --></p>
    </hgroup>
  </slide>

  
    <slide class="level-1">
    <hgroup>
      <h1>Problem Set 3<a class="headerlink" href="../../../Problem_Sets/problem_set_3/problem_set_3.html#problem-set-3" title="View HTML">§</a></h1>
    </hgroup>
    <article class="">
      <table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Due date:</th><td class="field-body">2014 Feb 21 at 9 PM MST</td>
</tr>
</tbody>
</table>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Overview<a class="headerlink" href="../../../Problem_Sets/problem_set_3/problem_set_3.html#overview" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>For this quiz you will write programs in Python to analyze data.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Continue to use the organization scheme that we learned about in
<a class="reference internal" href="../problem_set_1/problem_set_1.html#problem-set-1"><em>Problem Set 1</em></a>. Part of our evaluation
will include whether you are developing good organizational habits.</p>
</div>
<p>Create a <tt class="docutils literal"><span class="pre">run.sh</span></tt> file that executes the commands for each problem and
writes out each result in a dated directory.</p>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Problem 1 (BED files)<a class="headerlink" href="../../../Problem_Sets/problem_set_3/problem_set_3.html#problem-1-bed-files" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Use Python to perform a similar tasks as Problem Set 2, where you used
<tt class="docutils literal"><span class="pre">awk</span></tt>. Write a Python program to read the lamina.bed file and report the
following:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">What is the region with the largest start position (2nd column) <strong>for
each</strong> chromosome in <cite>lamina.bed</cite>? (<strong>5 points</strong>)</p>
</li>
<li><p class="first">What is the region with the largest end position on chrY in
lamina.bed? (<strong>5 points</strong>) Report as:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">chrom</span> <span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span> <span class="n">start</span> <span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span> <span class="n">end</span> <span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span> <span class="n">value</span> <span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span> <span class="n">region_length</span>
</pre></div>
</div>
</li>
</ol>
</div></blockquote>
<p>Download the bed file here: <a class="reference internal" href="../../misc/data.html#bed-file"><em>BED format</em></a></p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">See end of this page for the solution to question 1.</p>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Problem 2 (FASTQ files)<a class="headerlink" href="../../../Problem_Sets/problem_set_3/problem_set_3.html#problem-2-fastq-files" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Use Python to read a file in FASTQ format. FASTQ records are comprised of
4 sections; in this case each section will be on a unique line:</p>
<div class="highlight-python"><div class="highlight"><pre>@cluster_2:UMI_ATTCCG             # record name
TTTCCGGGGCACATAATCTTCAGCCGGGCGC   # DNA sequence
+                                 # empty line; starts with &#39;+&#39;
9C;=;=&lt;9@4868&gt;9:67AA&lt;9&gt;65&lt;=&gt;591   # phred-scaled quality scores
</pre></div>
</div>
<p>Download the fastq file here: <a class="reference internal" href="../../misc/data.html#fastq-file"><em>FASTQ format</em></a></p>
<p>Write a Python program to parse the FASTQ records and report the
following:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Which of the first 10 sequence records has the largest number of 'C'
residues in the sequence? Report its record name (<strong>5 points</strong>).</li>
<li>For each of the first 10 records, Covert each character in the
quality score to a number, and sum the numbers. Use <a class="reference external" href="http://docs.python.org/2.7/library/functions.html#ord" title="(in Python v2.7)"><tt class="xref py py-func docutils literal"><span class="pre">ord()</span></tt></a>
to convert characters to numbers (<strong>5 points</strong>).</li>
<li>Use the Python <a class="reference external" href="http://docs.python.org/2.7/library/collections.html#collections.Counter" title="(in Python v2.7)"><tt class="xref py py-class docutils literal"><span class="pre">Counter</span></tt></a> to count unique
sequences in the Fastq file. Report the top ten most abundant
sequences in the Fastq file (<strong>5 points</strong>).</li>
<li>Report the revese complement of each of the first 10 sequences (<strong>5
points</strong>)</li>
</ol>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The reverse complement of <tt class="docutils literal"><span class="pre">5'-AGCTCGTA-3''</span></tt> is <tt class="docutils literal"><span class="pre">5'-TACGAGCT-3'</span></tt></p>
</div>




    </article>
  </slide>  <slide class="level-2">
    <hgroup>
      <h2>Problem Set Submission<a class="headerlink" href="../../../Problem_Sets/problem_set_3/problem_set_3.html#problem-set-submission" title="View HTML">§</a></h2>
    </hgroup>
    <article class="">
      <p>Submit your problem set as a tar file to Canvas
(<a class="reference internal" href="../../Problem_Sets.html#problem-set-submission"><em>Problem Set Submission</em></a>).</p>




    </article>
  </slide>  <slide class="level-3">
    <hgroup>
      <h3>Question 1<a class="headerlink" href="../../../Problem_Sets/problem_set_3/problem_set_3.html#question-1" title="View HTML">§</a></h3>
    </hgroup>
    <article class="">
      <p>Here is one way to answer question 1. Note that we could also take
advantage of the fact that the file is sorted so the largest start
will always be the last line for each chromosome. Instead, here we
store the largest start we've seen so far, along with the value
in a dictionary keyed by chromosome.</p>
<div class="highlight-python"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</pre></div></td><td class="code"><div class="highlight"><pre><span class="kn">import</span> <span class="nn">sys</span>

<span class="n">bedfile</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>

<span class="n">largest_by_chrom</span> <span class="o">=</span> <span class="p">{}</span>

<span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">bedfile</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">line</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s">&#39;#&#39;</span><span class="p">):</span> <span class="k">continue</span>
    <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">value</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">rstrip</span><span class="p">(</span><span class="s">&#39;</span><span class="se">\r\n</span><span class="s">&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s">&#39;</span><span class="se">\t</span><span class="s">&#39;</span><span class="p">)</span>
    <span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">start</span><span class="p">)</span>

    <span class="c"># hint for question 2. if we wanted to just get chrY, we could add...</span>
    <span class="c"># if chrom != &quot;chrY&quot;: continue</span>

    <span class="c"># we haven&#39;t seen it before, so it has to be the largest</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">chrom</span> <span class="ow">in</span> <span class="n">largest_by_chrom</span><span class="p">:</span>
        <span class="n">largest_by_chrom</span><span class="p">[</span><span class="n">chrom</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">value</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="c"># we have to check if the current start is greater</span>
        <span class="c"># than the one we&#39;ve stored.</span>
        <span class="n">largest_chrom</span><span class="p">,</span> <span class="n">largest_start</span><span class="p">,</span> <span class="n">largest_end</span><span class="p">,</span> <span class="n">largest_value</span> \
                                               <span class="o">=</span> <span class="n">largest_by_chrom</span><span class="p">[</span><span class="n">chrom</span><span class="p">]</span>
        <span class="k">if</span> <span class="n">start</span> <span class="o">&gt;</span> <span class="n">largest_start</span><span class="p">:</span>
            <span class="c"># if it is, then we store a new region</span>
            <span class="n">largest_by_chrom</span><span class="p">[</span><span class="n">chrom</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">value</span><span class="p">)</span>

<span class="c"># see what it looks like:</span>
<span class="c"># print largest_by_chrom</span>
<span class="k">for</span> <span class="n">chrom</span> <span class="ow">in</span> <span class="n">largest_by_chrom</span><span class="p">:</span>
    <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">value</span> <span class="o">=</span> <span class="n">largest_by_chrom</span><span class="p">[</span><span class="n">chrom</span><span class="p">]</span>
    <span class="k">print</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">value</span>
</pre></div>
</td></tr></table></div>




    </article>
  </slide>


    <slide class="thank-you-slide segue nobackground">
    <aside class="gdbar right"><img src="images/google_developers_icon_128.png"></aside>
    <article class="flexbox vleft auto-fadein">
      <h2>&lt;Thank You!&gt;</h2>
    </article>
    <p class="auto-fadein" data-config-contact>
      <!-- populated from slide_config.json -->
    </p>
  </slide>

  <slide class="backdrop"></slide>

</slides>

<!--[if IE]>
  <script src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js"></script>
  <script>CFInstall.check({mode: 'overlay'});</script>
<![endif]-->
</body>
</html>