

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Class 25 : Exome-Seq - Alignment &mdash; Genome Informatics Workshop 0.1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  
    <link rel="top" title="Genome Informatics Workshop 0.1 documentation" href="../index.html"/>
        <link rel="up" title="Class list" href="../Classes.html"/>
        <link rel="next" title="Class 26 : Exome-Seq - Variants" href="class-26.html"/>
        <link rel="prev" title="Class 24 : RNA-Seq II" href="class-24.html"/> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>



  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="../index.html" class="fa fa-home"> Genome Informatics Workshop</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../Syllabus.html">Syllabus (MOLB FIXME)</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal" href="../Classes.html">Class list</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="class-1.html">Class 1 : Virtual Machine (VM) and Linux Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-2.html">Class 2 : The command-line</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-3.html">Class 3 : The command-line (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-4.html">Class 4 : awk</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-5.html">Class 5 : Working in a cluster environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-6.html">Class 6 : Working in a cluster environment (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-7.html">Class 7 : Python : Basics</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-8.html">Class 8 : Python : Basics (2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-9.html">Class 9 : Python Concepts</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-10.html">Class 10 : Applied Python</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-11.html">Class 11 : Python Idioms</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-12.html">Class 12 : The ENCODE Project</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-13.html">Class 13 : BEDtools</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-14.html">Class 14 : ChIP-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-15.html">Class 15 : ChIP-seq / DNase I</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-16.html">Class 16: Genomic Analysis Vignettes</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-17.html">Class 17 : Bash and Bam</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-18.html">Class 18 : R : Getting Started</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-19.html">Class 19 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-20.html">Class 20 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-21.html">Class 21 : R : DataFrames &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-22.html">Class 22 : R : Data Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-23.html">Class 23 : RNA-Seq I</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-24.html">Class 24 : RNA-Seq II</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="">Class 25 : Exome-Seq - Alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="class-26.html">Class 26 : Exome-Seq - Variants</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../Problem_Sets.html">Problem Sets</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_1/problem_set_1.html">Problem Set 1</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_2/problem_set_2.html">Problem Set 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_3/problem_set_3.html">Problem Set 3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_4/problem_set_4.html">Problem Set 4</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_5/problem_set_5.html">Problem Set 5</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/problem_set_6/problem_set_6.html">Problem Set 6</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Problem_Sets/Problem_Set_Keys.html">Problem Set : KEYS</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../Miscellaneous.html">Miscellaneous</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../misc/commands.html">Reference: list of commands</a></li>
<li class="toctree-l2"><a class="reference internal" href="../misc/data.html">Sample Data files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../misc/tips.html">Programming Tips &amp; Tricks</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">Genome Informatics Workshop</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
          <li><a href="../Classes.html">Class list</a> &raquo;</li>
      
    <li>Class 25 : Exome-Seq - Alignment</li>
      <li class="wy-breadcrumbs-aside">
        
          <a href="../_sources/Classes/class-25.txt" rel="nofollow"> View page source</a>
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            
  <div class="section" id="class-25-exome-seq-alignment">
<h1>Class 25 : Exome-Seq - Alignment<a class="headerlink" href="#class-25-exome-seq-alignment" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#class-25-exome-seq-alignment" title="Slides">§</a></h1>
<div class="section" id="goals">
<h2>Goals<a class="headerlink" href="#goals" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#goals" title="Slides">§</a></h2>
<blockquote>
<div><ol class="arabic simple">
<li>Not kill Tesla</li>
<li>Exome Sequencing</li>
<li>Alignment</li>
<li>QC Alignments</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="exome-sequencing">
<h2>Exome Sequencing<a class="headerlink" href="#exome-sequencing" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#exome-sequencing" title="Slides">§</a></h2>
<blockquote>
<div><ul class="simple">
<li>captures coding regions of genome</li>
<li>deeper coverage and more samples at same cost</li>
<li>goal is to call variants (SNPs/deletions/insertions)</li>
<li>higher concentration of interesting variants
relative to non-coding genome</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="project-design">
<h2>Project Design<a class="headerlink" href="#project-design" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#project-design" title="Slides">§</a></h2>
<img alt="../_images/phenotype.png" src="../_images/phenotype.png" />
</div>
<div class="section" id="project-design-ii">
<h2>Project Design II<a class="headerlink" href="#project-design-ii" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#project-design-ii" title="Slides">§</a></h2>
<p>8 sequenced samples from single family</p>
<blockquote>
<div><ul class="simple">
<li>paired-end 100 base reads</li>
<li>5 affected</li>
<li>3 unaffected</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="analysis">
<h2>Analysis<a class="headerlink" href="#analysis" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#analysis" title="Slides">§</a></h2>
<blockquote>
<div><ul class="simple">
<li>How likely are we to find something?</li>
<li>Naive probability that 5 individuals have a variant and 3 do not (without
considering pedigree)?</li>
<li>What if we find 1 million SNPs total?</li>
<li>How can we filter?</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="alignment">
<h2>Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#alignment" title="Slides">§</a></h2>
<blockquote>
<div><ul class="simple">
<li>bwa mem to align the reads.</li>
<li>bwa mem very good aligner for long reads</li>
</ul>
<p><a class="reference external" href="http://arxiv.org/pdf/1303.3997v2.pdf">http://arxiv.org/pdf/1303.3997v2.pdf</a></p>
</div></blockquote>
</div>
<div class="section" id="bwa-mem">
<h2>BWA mem<a class="headerlink" href="#bwa-mem" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#bwa-mem" title="Slides">§</a></h2>
<img alt="../_images/bwa-mem-roc.png" src="../_images/bwa-mem-roc.png" />
</div>
<div class="section" id="mark-duplicates">
<h2>Mark Duplicates<a class="headerlink" href="#mark-duplicates" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#mark-duplicates" title="Slides">§</a></h2>
<p>Marking/removing duplicates is to remove the influence of PCR duplication on
variant calling.</p>
<blockquote>
<div><ul class="simple">
<li>Marking/Removing duplicates makes sense unless you have extremely deep coverage,
especially on paired-end reads.</li>
<li>Uses position of both reads of a pair to determine duplication&#8211;if many pairs
have the same start and sequence, all but 1 pair is marked as a duplicate.</li>
<li>Recommended by the variant caller that we will use.</li>
<li>To mark or not to mark? <a class="reference external" href="http://seqanswers.com/forums/showthread.php?t=6854">http://seqanswers.com/forums/showthread.php?t=6854</a></li>
</ul>
</div></blockquote>
</div>
<div class="section" id="read-groups">
<h2>Read Groups<a class="headerlink" href="#read-groups" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#read-groups" title="Slides">§</a></h2>
<p>Your alignments should have read-groups.</p>
<p>These tell downstream tools what library and sample your data came from</p>
<p>Used for calling duplicates (repeats must be from same library)</p>
<p>With read-groups, can have multiple samples in a single BAM.</p>
</div>
<div class="section" id="variant-calling">
<h2>Variant Calling<a class="headerlink" href="#variant-calling" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#variant-calling" title="Slides">§</a></h2>
<p>At each site in the genome, given:</p>
<blockquote>
<div><ul class="simple">
<li>base calls</li>
<li>base qualities</li>
<li>mapping qualities</li>
<li>surrounding sequence</li>
<li>position in read</li>
<li>strand bias</li>
<li>all of the above for other samples</li>
</ul>
</div></blockquote>
<p>What is the genotype of each sample and the confidence in that genotype?</p>
</div>
<div class="section" id="variant-calling-software">
<h2>Variant Calling Software<a class="headerlink" href="#variant-calling-software" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#variant-calling-software" title="Slides">§</a></h2>
<p>The most commonly used software for variant calling is GATK:
Genome Analysis Toolkit</p>
<p>It has a number of steps including base recalibration, indel realignment,
etc.</p>
<p>We will use <cite>freebayes</cite> which has been shown to perform as well/better
(and it is much easier to use): <a class="reference external" href="https://github.com/ekg/freebayes">https://github.com/ekg/freebayes</a></p>
<p><a class="reference external" href="http://bcbio.wordpress.com/2013/10/21/updated-comparison-of-variant-detection-methods-ensemble-freebayes-and-minimal-bam-preparation-pipelines/">http://bcbio.wordpress.com/2013/10/21/updated-comparison-of-variant-detection-methods-ensemble-freebayes-and-minimal-bam-preparation-pipelines/</a></p>
</div>
<div class="section" id="freebayes">
<h2>Freebayes<a class="headerlink" href="#freebayes" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#freebayes" title="Slides">§</a></h2>
<p><a class="reference external" href="https://github.com/ekg/freebayes">https://github.com/ekg/freebayes</a></p>
<p><em>FreeBayes is haplotype-based, in the sense that it calls variants based on the
literal sequences of reads aligned to a particular target, not their precise
alignment.</em></p>
<p><em>FreeBayes uses short-read alignments (BAM files) for any number of individuals
from a population and a reference genome to determine the most-likely combination
of genotypes for the population at each position in the reference. It reports
positions which it finds putatively polymorphic in variant call file (VCF) format.</em></p>
</div>
<div class="section" id="vcf-format">
<h2>VCF Format<a class="headerlink" href="#vcf-format" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#vcf-format" title="Slides">§</a></h2>
<p>File format for SNP/indel calls:
<a class="reference external" href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41">http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41</a></p>
<div class="highlight-python"><div class="highlight"><pre>CHROM      1
POS        2345544
ID         rs1234..
REF        C
ALT        TA
QUAL       55.2
FILTER     PASS
INFO       NS=3;DP=14;AF=0.5;DB
FORMAT     GT:GQ:DP:HQ
sample_1   0|0:48:1:51,51
...
sample_n
</pre></div>
</div>
<p>&#8221;:&#8221; delimited chunks from samples match the format in <cite>FORMAT</cite></p>
</div>
<div class="section" id="pipeline">
<h2>Pipeline<a class="headerlink" href="#pipeline" title="Permalink to this headline">¶</a><a class="headerlink" href="../slides/Classes/class-25.html#pipeline" title="Slides">§</a></h2>
<p>Walk through .sh scripts on <em>amc-tesla</em></p>
</div>
</div>


          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="class-26.html" class="btn btn-neutral float-right" title="Class 26 : Exome-Seq - Variants"/>Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="class-24.html" class="btn btn-neutral" title="Class 24 : RNA-Seq II"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2014, Brent Pedersen, Jay Hesselberth.
      Last updated on Aug 24, 2014.
    </p>
  </div>

  <a href="https://github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'0.1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>