

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Problem Set 6 : KEY &mdash; Genome Informatics Workshop 0.1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  

    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:'../../',
        VERSION:'0.1',
        COLLAPSE_INDEX:false,
        FILE_SUFFIX:'.html',
        HAS_SOURCE:  true
      };
    </script>
      <script type="text/javascript" src="../../_static/jquery.js"></script>
      <script type="text/javascript" src="../../_static/underscore.js"></script>
      <script type="text/javascript" src="../../_static/doctools.js"></script>
      <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

    

  

  
  
    <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
    <link rel="top" title="Genome Informatics Workshop 0.1 documentation" href="../../index.html"/>
        <link rel="up" title="Problem Set : KEYS" href="../Problem_Set_Keys.html"/>
        <link rel="next" title="Miscellaneous" href="../../Miscellaneous.html"/>
        <link rel="prev" title="Problem Set 5 : KEY" href="problem-set-5-key.html"/> 

<!-- Google Analytics tracking code -->
<!--        DO NOT REMOVE           -->
<!--           JRH                  -->

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-47837611-1', 'ucd-bioworkshop.github.io');
  ga('send', 'pageview');

</script>



  <script src="//cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="../../index.html" class="icon icon-home"> Genome Informatics Workshop</a>
        <form class="wy-form" action="../../search.html" method="get">
  <input type="text" name="q" placeholder="Search docs" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../Syllabus.html">Syllabus (MOLB FIXME)</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../Classes.html">Class list</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-1.html">Class 1 : Virtual Machine (VM) and Linux Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-2.html">Class 2 : The command-line</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-3.html">Class 3 : The command-line (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-4.html">Class 4 : awk</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-5.html">Class 5 : Working in a cluster environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-6.html">Class 6 : Working in a cluster environment (part 2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-7.html">Class 7 : Python : Basics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-8.html">Class 8 : Python : Basics (2)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-9.html">Class 9 : Python Concepts</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-10.html">Class 10 : Applied Python</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-11.html">Class 11 : Python Idioms</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-12.html">Class 12 : The ENCODE Project</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-13.html">Class 13 : BEDtools</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-14.html">Class 14 : ChIP-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-15.html">Class 15 : ChIP-seq / DNase I</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-16.html">Class 16: Genomic Analysis Vignettes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-17.html">Class 17 : Bash and Bam</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-18.html">Class 18 : R : Getting Started</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-19.html">Class 19 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-20.html">Class 20 : R : Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-21.html">Class 21 : R : DataFrames &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-22.html">Class 22 : R : Data Manipulation &amp; Plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-23.html">Class 23 : RNA-Seq I</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-24.html">Class 24 : RNA-Seq II</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-25.html">Class 25 : Exome-Seq - Alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../Classes/class-26.html">Class 26 : Exome-Seq - Variants</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal" href="../../Problem_Sets.html">Problem Sets</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../problem_set_1/problem_set_1.html">Problem Set 1</a></li>
<li class="toctree-l2"><a class="reference internal" href="../problem_set_2/problem_set_2.html">Problem Set 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../problem_set_3/problem_set_3.html">Problem Set 3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../problem_set_4/problem_set_4.html">Problem Set 4</a></li>
<li class="toctree-l2"><a class="reference internal" href="../problem_set_5/problem_set_5.html">Problem Set 5</a></li>
<li class="toctree-l2"><a class="reference internal" href="../problem_set_6/problem_set_6.html">Problem Set 6</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../Problem_Set_Keys.html">Problem Set : KEYS</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../Miscellaneous.html">Miscellaneous</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../misc/commands.html">Reference: list of commands</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../misc/data.html">Sample Data files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../misc/tips.html">Programming Tips &amp; Tricks</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top">
        <i data-toggle="wy-nav-top" class="icon icon-reorder"></i>
        <a href="/">Genome Informatics Workshop</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <ul class="wy-breadcrumbs">
  <li><a href="../../index.html">Docs</a> &raquo;</li>
  <li><a href="">Problem Set 6 : KEY</a></li>
  
</ul>
<hr/>

          
  <div class="section" id="problem-set-6-key">
<span id="id1"></span><h1>Problem Set 6 : KEY<a class="headerlink" href="#problem-set-6-key" title="Permalink to this headline">¶</a><a class="headerlink" href="../../slides/Problem_Sets/problem-set-keys/problem-set-6-key.html#problem-set-6-key" title="Slides">§</a></h1>
<div class="section" id="problem-2-1">
<h2>Problem 2.1<a class="headerlink" href="#problem-2-1" title="Permalink to this headline">¶</a><a class="headerlink" href="../../slides/Problem_Sets/problem-set-keys/problem-set-6-key.html#problem-2-1" title="Slides">§</a></h2>
<p>The following R block calculates the range of each transcription factor
peak widths.</p>
<div class="highlight-r"><div class="highlight"><pre><span class="o">&gt;</span> library<span class="p">(</span>dplyr<span class="p">)</span>
<span class="o">&gt;</span> colnames <span class="o">&lt;-</span> c<span class="p">(</span><span class="s">&#39;chrom&#39;</span><span class="p">,</span><span class="s">&#39;start&#39;</span><span class="p">,</span><span class="s">&#39;end&#39;</span><span class="p">,</span><span class="s">&#39;name&#39;</span><span class="p">)</span>
<span class="o">&gt;</span> peaks_df <span class="o">&lt;-</span> read.table<span class="p">(</span><span class="s">&#39;/opt//bio-workshop//data//peaks.bed.gz&#39;</span><span class="p">,</span>
                          col.names<span class="o">=</span>colnames<span class="p">)</span>

<span class="o">&gt;</span> peaks_tbl <span class="o">&lt;-</span> tbl_df<span class="p">(</span>peaks_df<span class="p">)</span>

<span class="o">&gt;</span> summary <span class="o">&lt;-</span> peaks_tbl <span class="o">%.%</span>
<span class="o">+</span>    group_by<span class="p">(</span>name<span class="p">)</span> <span class="o">%.%</span>
<span class="o">+</span>    mutate<span class="p">(</span>peak.width <span class="o">=</span> end <span class="o">-</span> start<span class="p">)</span> <span class="o">%.%</span>
<span class="o">+</span>    summarize<span class="p">(</span>count <span class="o">=</span> n<span class="p">(),</span>
<span class="o">+</span>        min.width <span class="o">=</span> min<span class="p">(</span>peak.width<span class="p">),</span>
<span class="o">+</span>        max.width <span class="o">=</span> max<span class="p">(</span>peak.width<span class="p">))</span> <span class="o">%.%</span>
<span class="o">+</span>    mutate<span class="p">(</span>range <span class="o">=</span> max.width <span class="o">-</span> min.width<span class="p">)</span> <span class="o">%.%</span>
<span class="o">+</span>    arrange<span class="p">(</span>desc<span class="p">(</span>range<span class="p">))</span>
</pre></div>
</div>
<p>Now we want to grab the top ten factors ordered by range and make box and
violin plots.</p>
<div class="highlight-r"><div class="highlight"><pre><span class="o">&gt;</span> library<span class="p">(</span>ggplot2<span class="p">)</span>
<span class="o">&gt;</span> factors.top10 <span class="o">&lt;-</span> summary<span class="o">$</span>name<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">]</span>

<span class="c1"># the subset function returns parts of a data.frame, in this case it</span>
<span class="c1"># returns rows where name is in the factors data.frame</span>
<span class="o">&gt;</span> peaks.subset <span class="o">&lt;-</span> subset<span class="p">(</span>peaks_tbl<span class="p">,</span> name <span class="o">%in%</span> factors.top10<span class="p">)</span>

<span class="c1"># end - start = width</span>
<span class="o">&gt;</span> gp <span class="o">&lt;-</span> ggplot<span class="p">(</span>peaks.subset<span class="p">,</span> aes<span class="p">(</span>name<span class="p">,</span> end <span class="o">-</span> start<span class="p">,</span> color<span class="o">=</span>name<span class="p">))</span>
<span class="o">&gt;</span> gp <span class="o">+</span> geom_boxplot<span class="p">()</span> <span class="o">+</span> scale_y_log10<span class="p">()</span> <span class="o">+</span> coord_flip<span class="p">()</span>

<span class="c1"># save the plot</span>
ggsave<span class="p">(</span><span class="s">&#39;problem-2-1-boxplot.pdf&#39;</span><span class="p">)</span>

<span class="c1"># or use geom_violin()</span>
<span class="o">&gt;</span> gp <span class="o">+</span> geom_violin<span class="p">()</span> <span class="o">+</span> coord_flip<span class="p">()</span>

<span class="c1"># save the plot</span>
ggsave<span class="p">(</span><span class="s">&#39;problem-2-1-violin.pdf&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="problem-2-2">
<h2>Problem 2.2<a class="headerlink" href="#problem-2-2" title="Permalink to this headline">¶</a><a class="headerlink" href="../../slides/Problem_Sets/problem-set-keys/problem-set-6-key.html#problem-2-2" title="Slides">§</a></h2>
<p>Grab the peaks that do not overlap any other peaks (besides itself). Strategy:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Use <a class="reference external" href="http://bedtools.readthedocs.org/en/latest/content/tools/intersect.html#intersect" title="(in bedtools v2.20.1)"><em class="xref std std-ref">bedtools intersect</em></a></li>
<li>Give the same filename to <tt class="docutils literal"><span class="pre">-a</span></tt> and <tt class="docutils literal"><span class="pre">-b</span></tt></li>
<li>Count the number of overlaps with <tt class="docutils literal"><span class="pre">-c</span></tt></li>
<li>Use <tt class="docutils literal"><span class="pre">awk</span></tt> to select those regions with only 1 overlap (i.e. 1 means
the only overlap is with itself)</li>
</ol>
</div></blockquote>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ fname</span><span class="o">=</span><span class="s2">&quot;peaks.bed.gz&quot;</span>
<span class="nv">$ </span>bedtools intersect -a <span class="nv">$fname</span> -b <span class="nv">$fname</span> -c <span class="se">\</span>
    | awk <span class="s1">&#39;$NF == 1&#39;</span> <span class="se">\</span>
    | gzip -c <span class="se">\</span>
    &gt; peaks.no-overlap.bed.gz
</pre></div>
</div>
<p>Feed the new BED file into the pipeline for Problem 2.1 (above).</p>
</div>
</div>


          <footer>
  
    <div class="rst-footer-buttons">
      
        <a href="../../Miscellaneous.html" class="btn btn-neutral float-right" title="Miscellaneous"/>Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="problem-set-5-key.html" class="btn btn-neutral" title="Problem Set 5 : KEY"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <p>
      &copy; Copyright 2014, Brent Pedersen, Jay Hesselberth.
    Last updated on Aug 25, 2014.
  </p>

  <a href="https://www.github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="http://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  

</body>
</html>